--- /src/defines_orig.h	2008-10-14 11:05:37.000000000 +0200
+++ /src/defines.h	2008-10-14 11:06:42.000000000 +0200
@@ -49,10 +49,11 @@
 
 /***** Various stuff that you may want to tweak, but probably shouldn't *****/
 
-#define SOCKETBUF_SIZE                          32768
-#define CLIENT_STREAM_SIZE                      8192
-#define BUFFER_SIZE                             4096
-#define MAX_HEADER_LENGTH			1024
+/* Modified to allow a larger number of values transmitted through a single GET */
+#define SOCKETBUF_SIZE                          65536
+#define CLIENT_STREAM_SIZE                      16384
+#define BUFFER_SIZE                             8192
+#define MAX_HEADER_LENGTH			4096
 
 #define MIME_HASHTABLE_SIZE			47
 #define ALIAS_HASHTABLE_SIZE                    17
