diff --exclude CVS -uNr can4linux-examples/can_send.c can4linux-examples.modified/can_send.c
--- can4linux-examples/can_send.c	2005-02-04 09:37:28.000000000 -0700
+++ can4linux-examples.modified/can_send.c	2006-05-08 16:02:28.982582816 -0600
@@ -223,6 +223,7 @@
 		load = atoi(optarg);
 		doload = TRUE;
 		break;
+#ifdef USE_RT_SCHEDULING
 	    case 'p':
 	        {
 	        struct sched_param mysched;
@@ -234,7 +235,6 @@
 		      fprintf(stderr, "Priority > %d not allowed\n",
 		      					max_priority);
 		    }
-#ifdef USE_RT_SCHEDULING
 		    mysched.sched_priority =
 		    		sched_get_priority_max(SCHED_RR) - 1;
 		    ret = sched_setscheduler(0,SCHED_FIFO,&mysched);
@@ -247,9 +247,9 @@
 		    if ( debug == TRUE ) {
 			printf("mlockall() = %d\n", ret);
 		    }
-#endif
 		}
 		break;
+#endif
 	    case 's':
 		sleeptime = atoi(optarg);
 		break;
