diff --exclude CVS --exclude .git -uNr linux-2.6.35.3/drivers/net/fec.c linux-2.6.35.3.modified/drivers/net/fec.c
--- linux-2.6.35.3/drivers/net/fec.c	2012-11-09 08:40:20.727521449 +0100
+++ linux-2.6.35.3.modified/drivers/net/fec.c	2012-10-31 09:08:54.550208441 +0100
@@ -1269,12 +1269,18 @@
 fec_restart(struct net_device *dev, int duplex)
 {
 	struct fec_enet_private *fep = netdev_priv(dev);
+	struct fec_platform_data *pdata = fep->pdev->dev.platform_data;
 	int i;
 	uint ret = 0;
 	u32 temp_mac[2];
 	unsigned long reg;
 	int val;
 
+#ifdef CONFIG_ARCH_MXS
+	if (pdata && pdata->init)
+		ret = pdata->init();
+#endif
+
 	/* Whack a reset.  We should wait for this. */
 	writel(1, fep->hwp + FEC_ECNTRL);
 	udelay(10);
