diff --exclude CVS --exclude .git -uNr linux-2.6.35.3/arch/arm/mach-mx28/mx28evk.c linux-2.6.35.3.modified/arch/arm/mach-mx28/mx28evk.c
--- linux-2.6.35.3/arch/arm/mach-mx28/mx28evk.c	2013-07-26 10:01:44.993313001 +0200
+++ linux-2.6.35.3.modified/arch/arm/mach-mx28/mx28evk.c	2013-06-27 10:41:26.000000000 +0200
@@ -58,16 +58,28 @@
 #endif
 
 static struct spi_board_info spi_board_info[] __initdata = {
+#if (defined(CONFIG_MTD_M25P80) || defined(CONFIG_MTD_M25P80_MODULE)) && (defined(CONFIG_RTC_DRV_DS1390) || defined(CONFIG_RTC_DRV_DS1390_MODULE))
+#error Invalid kernel configuration: CONFIG_MTD_M25P80 and CONFIG_RTC_DRV_DS1390 use the same chipselect.
+#endif
 #if defined(CONFIG_MTD_M25P80) || defined(CONFIG_MTD_M25P80_MODULE)
 	{
 		/* the modalias must be the same as spi device driver name */
 		.modalias = "m25p80", /* Name of spi_driver for this device */
 		.max_speed_hz = 20000000,     /* max spi clock (SCK) speed in HZ */
 		.bus_num = 1, /* Framework bus number */
-		.chip_select = 0, /* Framework chip select. */
+		.chip_select = 0, /* Framework chip select.*/
 		.platform_data = &mx28_spi_flash_data,
 	},
 #endif
+#if defined(CONFIG_RTC_DRV_DS1390) || defined(CONFIG_RTC_DRV_DS1390_MODULE)
+	{
+		.modalias = "rtc-ds1390",
+		.max_speed_hz = 921600, /* run SSP as fast as possbile wrt DS1390 timing constraint*/
+		.bus_num = 1, /* Framework bus number */
+		.chip_select = 0, /* Framework chip select. SSP2_SS0*/
+		//.mode = SPI_MODE_3, /* Do not overwrite the standard configuration otherwise spi  won't setup */
+	},
+#endif
 };
 
 static void spi_device_init(void)
