diff --exclude CVS --exclude .git -uNr sftp-server-4.3p2/openssh-4.3p2/Makefile.in sftp-server-4.3p2.modified/openssh-4.3p2/Makefile.in
--- sftp-server-4.3p2/openssh-4.3p2/Makefile.in	2006-01-01 00:47:05.000000000 -0800
+++ sftp-server-4.3p2.modified/openssh-4.3p2/Makefile.in	2012-03-08 17:28:28.610248331 -0800
@@ -157,7 +157,7 @@
 	$(LD) -o $@ ssh-keyscan.o $(LDFLAGS) -lssh -lopenbsd-compat -lssh $(LIBS)
 
 sftp-server$(EXEEXT): $(LIBCOMPAT) libssh.a sftp.o sftp-common.o sftp-server.o
-	$(LD) -o $@ sftp-server.o sftp-common.o $(LDFLAGS) -lssh -lopenbsd-compat $(LIBS)
+	$(LD) -o $@ sftp-server.o sftp-common.o $(LDFLAGS) -Wl,-Bstatic -lssh -lopenbsd-compat -lcrypto -Wl,-Bdynamic 
 
 sftp$(EXEEXT): $(LIBCOMPAT) libssh.a sftp.o sftp-client.o sftp-common.o sftp-glob.o progressmeter.o
 	$(LD) -o $@ progressmeter.o sftp.o sftp-client.o sftp-common.o sftp-glob.o $(LDFLAGS) -lssh -lopenbsd-compat $(LIBS) $(LIBEDIT)
@@ -234,6 +234,10 @@
 install-nokeys: $(CONFIGFILES) ssh_prng_cmds.out $(MANPAGES) $(TARGETS) install-files install-sysconf
 install-nosysconf: $(CONFIGFILES) ssh_prng_cmds.out $(MANPAGES) $(TARGETS) install-files
 
+install-sftp-server:
+	$(srcdir)/mkinstalldirs $(DESTDIR)$(libexecdir)
+	$(INSTALL) -m 0755 $(STRIP_OPT) sftp-server $(DESTDIR)$(SFTP_SERVER)
+
 check-config:
 	-$(DESTDIR)$(sbindir)/sshd -t -f $(DESTDIR)$(sysconfdir)/sshd_config
 
