diff --exclude CVS --exclude .git -uNr qt-everywhere-opensource-src-4.8.5/config.tests/unix/mysql/mysql.cpp qt-everywhere-opensource-src-4.8.5.modified/config.tests/unix/mysql/mysql.cpp
--- qt-everywhere-opensource-src-4.8.5/config.tests/unix/mysql/mysql.cpp	2013-06-07 07:16:41.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.5.modified/config.tests/unix/mysql/mysql.cpp	2015-01-09 16:11:00.922822005 +0100
@@ -39,7 +39,7 @@
 **
 ****************************************************************************/
 
-#include "mysql.h"
+#include "mysql/mysql.h"
 
 int main(int, char **)
 {
diff --exclude CVS --exclude .git -uNr qt-everywhere-opensource-src-4.8.5/src/sql/drivers/mysql/qsql_mysql.h qt-everywhere-opensource-src-4.8.5.modified/src/sql/drivers/mysql/qsql_mysql.h
--- qt-everywhere-opensource-src-4.8.5/src/sql/drivers/mysql/qsql_mysql.h	2013-06-07 07:17:00.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.5.modified/src/sql/drivers/mysql/qsql_mysql.h	2015-01-09 16:25:55.765020004 +0100
@@ -49,7 +49,7 @@
 #include <QtCore/qt_windows.h>
 #endif
 
-#include <mysql.h>
+#include <mysql/mysql.h>
 
 #ifdef QT_PLUGIN
 #define Q_EXPORT_SQLDRIVER_MYSQL
