diff --exclude CVS --exclude .git -uNr qwt-6.0.1/qwt.pro qwt-6.0.1.modified/qwt.pro
--- qwt-6.0.1/qwt.pro	2011-08-01 16:33:53.000000000 +0200
+++ qwt-6.0.1.modified/qwt.pro	2012-10-18 09:01:08.327574546 +0200
@@ -16,13 +16,7 @@
     src \
     textengines
 
-contains(QWT_CONFIG, QwtDesigner ) {
-    SUBDIRS += designer 
-}
 
-contains(QWT_CONFIG, QwtExamples ) {
-    SUBDIRS += examples 
-}
 
 qwtspec.files  = qwtconfig.pri qwt.prf
 qwtspec.path  = $${QWT_INSTALL_FEATURES}
diff --exclude CVS --exclude .git -uNr qwt-6.0.1/qwt.pro.ori qwt-6.0.1.modified/qwt.pro.ori
--- qwt-6.0.1/qwt.pro.ori	1970-01-01 01:00:00.000000000 +0100
+++ qwt-6.0.1.modified/qwt.pro.ori	2012-10-18 09:01:08.299574546 +0200
@@ -0,0 +1,31 @@
+################################################################
+# Qwt Widget Library
+# Copyright (C) 1997   Josef Wilgen
+# Copyright (C) 2002   Uwe Rathmann
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the Qwt License, Version 1.0
+################################################################
+
+include( qwtconfig.pri )
+
+TEMPLATE = subdirs
+CONFIG   += ordered
+
+SUBDIRS = \
+    src \
+    textengines
+
+contains(QWT_CONFIG, QwtDesigner ) {
+    SUBDIRS += designer 
+}
+
+contains(QWT_CONFIG, QwtExamples ) {
+    SUBDIRS += examples 
+}
+
+qwtspec.files  = qwtconfig.pri qwt.prf
+qwtspec.path  = $${QWT_INSTALL_FEATURES}
+
+INSTALLS += qwtspec
+
