diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/S10setup skell-1.18.modified/etc/rc.d/init.d/S10setup
--- skell-1.18/etc/rc.d/init.d/S10setup	2015-04-02 16:32:18.379976976 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/S10setup	2015-03-30 14:10:10.286771706 +0200
@@ -11,9 +11,6 @@
 
 [ -d $mntdir ] || { mntdir=/tmp/mnt; /bin/mkdir $mntdir; }
 [ -d $mntdir ] || exit 1
-sleep 2
-dmesg_log=/tmp/log
-dmesg > $dmesg_log
 logfile=$mntdir/update.log
 UPDATE_DESCR_FILE=/local/kindofupdate.lck
   
@@ -21,46 +18,14 @@
  
 	rm -f $UPDATE_DESCR_FILE
     echo -n "Checking for configuration USB key...  "
-    sleep 1
     echo "" | tee $logfile
-	usb_string=`awk '/sd[a-z]: sd[a-z]/ ' < $dmesg_log`
-	if [ -z "$usb_string" ]; then
-		usb_string=`awk '/sd[a-z]:/ ' < $dmesg_log`
-	fi
-
-	echo "Found scsi device: $usb_string" | tee -a $logfile
-
-	if [ -n "$usb_string" ]; then
-		mntpoint=`echo $usb_string |  awk '{split($0,a,":"); print a[1]'}`
-		#echo "tentative dev=$mntpoint"
-		if /bin/mount -orw /dev/$mntpoint $mntdir 2>/dev/null; then
-			usbdev=/dev/$mntpoint
+	if [ -e /sys/block/sda/sda1 ]
+	then
+		echo "Found scsi device: $usb_string" | tee -a $logfile
+		if /bin/mount -orw /dev/sda1 $mntdir 2>/dev/null; then
+			usbdev=/dev/sda1
 			echo "current mount point is $usbdev" | tee -a $logfile
-		else
-			#echo "going to loop...."
-			usb_ok=0		
-			for i in 2 3 4 5 6 7 8 9
-			do
-				#echo "loop number $i"
-				if [ $usb_ok -eq 0 ]; then
-					mntpoint=`echo $usb_string |  awk '{split($0,a," "); print a['$i']'}`
-					#echo $mntpoint
-					if [ -n "$mntpoint" ]; then
-						if /bin/mount -orw /dev/$mntpoint $mntdir 2>/dev/null; then
-							usbdev=/dev/$mntpoint
-							usb_ok=1
-							echo "mounted scsi device in $mntpoint" | tee -a $logfile
-						fi
-					else
-						echo "no valid usb mountpoint found" | tee -a $logfile
-						break;
-					fi
-				else
-					break;
-				fi
-			done
 		fi
-
 	else
 		echo "no usb key found in dmesg" | tee -a $logfile
 		#echo "exit from configuration script" 	| tee -a $logfile
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/autoexec skell-1.18.modified/etc/rc.d/init.d/autoexec
--- skell-1.18/etc/rc.d/init.d/autoexec	2015-04-02 16:32:18.383978985 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/autoexec	2015-03-30 14:04:26.799151965 +0200
@@ -84,11 +84,11 @@
 elif [ -x $LOCAL_ROOT_DIR/fcrts ]
 then
 	echo "Starting applications."
-	dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
+	#dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
 
 	$LOCAL_ROOT_DIR/fcrts > /dev/null 2>&1 & 
 	if [ -x $LOCAL_ROOT_DIR/safe_hmi ]; then 
-		sleep 2
+		#sleep 2
 		$LOCAL_ROOT_DIR/safe_hmi > /dev/null 2>&1 &
 	fi
 elif [ -x $LOCAL_ROOT_DIR/chooser ]
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/network skell-1.18.modified/etc/rc.d/init.d/network
--- skell-1.18/etc/rc.d/init.d/network	2015-04-02 16:32:18.379976976 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/network	2015-03-30 14:01:53.459153311 +0200
@@ -81,7 +81,7 @@
         echo "config screen, or directly in $CONFIGFILE."
         echo "IP address setup bypassed"
         echo ""
-        sleep 2
+        #sleep 2
     fi
 
     if [ "$SYSCFG_IFACE0" = "y" ]
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/settime skell-1.18.modified/etc/rc.d/init.d/settime
--- skell-1.18/etc/rc.d/init.d/settime	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/settime	2015-03-30 14:01:30.943151650 +0200
@@ -27,7 +27,7 @@
             echo "    date <mmddhhmnyyyy>"
             echo "    /sbin/hwclock -w"
             touch /etc/setttime_msg
-            sleep 2
+            #sleep 2
         fi
     fi
 fi
