diff --exclude CVS --exclude .git -uNr skell-1.18/etc/inittab skell-1.18.modified/etc/inittab
--- skell-1.18/etc/inittab	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/etc/inittab	2011-10-13 14:07:38.559502135 +0200
@@ -8,8 +8,9 @@
 ::sysinit:/etc/rc.d/rcS
 
 # what to do after startup
-::respawn:-/bin/sh
+#::respawn:-/bin/sh
 #::respawn:/sbin/getty -L ttyS0 115200 vt100
+::respawn:/sbin/getty -L ttyAM0 115200 vt100
 
 # one way of rebooting
 ::ctrlaltdel:/sbin/reboot
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/boa skell-1.18.modified/etc/rc.d/init.d/boa
--- skell-1.18/etc/rc.d/init.d/boa	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/boa	2011-10-13 12:46:03.547317181 +0200
@@ -1,18 +1,60 @@
-#!/bin/sh
+#! /bin/sh
 
-if [ ! -x /usr/sbin/boa ]
-then
-    exit 0
+INDEXFILE=/local/www/index.html
+CGIFILE=index.cgi
+if [ -f $INDEXFILE ]; then
+/bin/rm -f $INDEXFILE
 fi
 
-if [ "$1" = "stop" -o "$1" = "restart" ]
-then                                                                            
-    echo "Stopping the boa webserver: "
-    killall boa
-fi
+IP=`ifconfig  | grep 'inet addr:'| grep -v '127.0.0.1' | cut -d: -f2 | awk '{ print $1}'`
+IPCHECK=` echo $IP | sed "s/[0-9]\{1,3\}//g" `
+
+start() {
 
-if [ "$1" = "start" -o "$1" = "restart" ]
-then
-    echo "Starting the boa webserver: "
-    /usr/sbin/boa $BOA_ARGS
+echo "<html>" >> $INDEXFILE
+echo "<head>" >> $INDEXFILE
+echo "<meta HTTP-EQUIV=\"REFRESH\" content=\"0; url=http://$IP/$CGIFILE\">" >> $INDEXFILE
+echo "</head>" >> $INDEXFILE
+echo "</html>" >> $INDEXFILE
+
+if [ -x /usr/sbin/boa  ]; then
+	if [ ${IPCHECK} = "..."  ] 2>/dev/null; then
+		echo -n "Starting BOA webserver... "
+		/usr/sbin/boa
+		echo "Done."
+	
+	else
+		echo "CANNOT start BOA webserver invalid ip address $IP."
+	fi	
 fi
+}
+
+stop() {
+echo -n "Shutting down BOA webserver... "
+killall boa 2> /dev/null
+echo "Done."S
+}
+
+restart() {
+stop
+sleep 1
+start
+}
+
+case "$1" in
+start)
+    start
+;;
+stop)
+    stop
+;;
+restart)
+    restart
+;;
+*)
+  echo $"Usage: $0 {start|stop|restart}"
+        exit 1
+    ;;
+    
+esac
+
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/network skell-1.18.modified/etc/rc.d/init.d/network
--- skell-1.18/etc/rc.d/init.d/network	2011-10-19 15:23:20.860896002 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/network	2011-10-13 14:46:32.139160285 +0200
@@ -1,5 +1,5 @@
 #!/bin/sh
-
+	
 CONFIGFILE=/local/etc/sysconfig/net.conf
 
 if [ ! -x /sbin/ifconfig ]
@@ -25,6 +25,36 @@
         source $CONFIGFILE
     fi
 
+    # in case the config file contain only one network config, use it as config 0
+	if [ -z $IPADDR0 ] && [ -n $IPADDR ]
+	then
+		IPADDR0=$IPADDR
+	fi
+	if [ -z $SYSCFG_IFACE0 ] && [ -n $SYSCFG_IFACE ]
+	then
+		SYSCFG_IFACE0=$SYSCFG_IFACE
+	fi
+	if [ -z $INTERFACE0 ] && [ -n $INTERFACE ]
+	then
+		INTERFACE0=$INTERFACE
+	fi
+	if [ -z $NETMASK0 ] && [ -n $NETMASK ]
+	then
+		NETMASK0=$NETMASK
+	fi
+	if [ -z $GATEWAY0 ] && [ -n $GATEWAY ]
+	then
+		GATEWAY0=$GATEWAY
+	fi
+	if [ -z $NAMESERVER0 ] && [ -n $NAMESERVER ]
+	then
+		NAMESERVER0=$NAMESERVER
+	fi
+	if [ -n $BOOTPROTO ] && [ "$BOOTPROTO" = "[DHCP]" ]
+	then
+		IPADDR0="dhcp"
+	fi
+
     if [ -n "$NAMESERVER0" ]
     then
 	/bin/mount -oremount,rw /
@@ -96,57 +126,6 @@
             fi
         fi
     fi
-    if [ "$SYSCFG_IFACE2" = "y" ]
-    then
-        echo "Setting up networking on $INTERFACE2: "
-        if [ "$IPADDR2" = "dhcp" ]
-        then
-            $SYSCFG_DHCPC_CMD $INTERFACE2
-        else  
-            # non-dhcp network startup
-            ifconfig $INTERFACE2 $IPADDR2 netmask $NETMASK2
-
-            if [ -n "$GATEWAY2" ]
-            then 
-                echo "Adding static route for default gateway to $GATEWAY2: "
-                route add default gateway $GATEWAY2 $INTERFACE2
-            fi
-        fi
-    fi
-    if [ "$SYSCFG_IFACE3" = "y" ]
-    then
-        echo "Setting up networking on $INTERFACE3: "
-        if [ "$IPADDR3" = "dhcp" ]
-        then
-            $SYSCFG_DHCPC_CMD $INTERFACE3
-        else  
-            # non-dhcp network startup
-            ifconfig $INTERFACE3 $IPADDR3 netmask $NETMASK3
-
-            if [ -n "$GATEWAY3" ]
-            then 
-                echo "Adding static route for default gateway to $GATEWAY3: "
-                route add default gateway $GATEWAY3 $INTERFACE3
-            fi
-        fi
-    fi
-    if [ "$SYSCFG_IFACE4" = "y" ]
-    then
-        echo "Setting up networking on $INTERFACE4: "
-        if [ "$IPADDR4" = "dhcp" ]
-        then
-            $SYSCFG_DHCPC_CMD $INTERFACE4
-        else  
-            # non-dhcp network startup
-            ifconfig $INTERFACE4 $IPADDR4 netmask $NETMASK4
-
-            if [ -n "$GATEWAY4" ]
-            then 
-                echo "Adding static route for default gateway to $GATEWAY4: "
-                route add default gateway $GATEWAY4 $INTERFACE4
-            fi
-        fi
-    fi
     if [ -n "$NAMESERVER0" ]
     then
 	/bin/mount -oremount,ro /

