--- ltib/dist/lfs-5.1/busybox/busybox.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/busybox/busybox.spec	2015-10-12 10:46:10.706466000 +0200
@@ -3,22 +3,35 @@
 
 Summary         : A small executable that replaces many UNIX utilities
 Name            : busybox
-Version         : 1.20.2
+Version         : 1.15.0
 Release         : 1
 License         : GPL
 URL             : http://www.busybox.net/
 Vendor          : Freescale
 Packager        : Steve Papacharalambous/Stuart Hughes
 Group           : System Environment/Shells
-Source          : %{name}-%{version}.tar.bz2
+Source          : %{name}-%{version}.tar.gz
+Patch1          : %{name}-%{version}-getty-nobaud-1.patch
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed busybox for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+busybox package contents restricted to just the files that are
+needed at run time on the target.
+
 
 %Description
 %{summary}
 
 %Prep
 %setup
+%patch1 -p1
 
 %Build
 PKG_BUSYBOX_PRECONFIG=${PKG_BUSYBOX_PRECONFIG:-busybox.config}
@@ -71,10 +84,20 @@
 EOF
 chmod 644 $RPM_BUILD_ROOT/%{pfx}/%{base}/etc/busybox.conf
 
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed -n '
+    p
+' all_files.lst > rfs_files.lst
+
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
-%Files
+%Files -f all_files.lst
+%defattr(-,root,root)
+%attr(4755,root,root) %{pfx}/%{base}/bin/busybox
+
+
+%Files rfs -f rfs_files.lst
 %defattr(-,root,root)
 %attr(4755,root,root) %{pfx}/%{base}/bin/busybox
-%{pfx}/*
