--- ltib/dist/lfs-5.1/bzip2/bzip2.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/bzip2/bzip2.spec	2015-10-12 10:46:10.710466000 +0200
@@ -15,6 +15,17 @@
 Patch2          : %{name}-%{version}-armv7a-cross-compile-fix.patch
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed bzip2 for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+bzip2 package contents restricted to just the files that are
+needed at run time on the target.
+
 
 %Description
 %{summary}
@@ -32,14 +43,23 @@
 
 %Install
 rm -rf $RPM_BUILD_ROOT
-make install PREFIX=$RPM_BUILD_ROOT/%{pfx}/%{_prefix}
+make install PREFIX=$RPM_BUILD_ROOT%{pfx}%{_prefix}
+
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_prefix}/share/man, d;
+    \,^%{pfx}%{_prefix}/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
 
 
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
