diff -ruN dropbear-2016.73/common-channel.c dropbear-2016.73.modified/common-channel.c
--- dropbear-2016.73/common-channel.c	2016-03-18 15:44:43.000000000 +0100
+++ dropbear-2016.73.modified/common-channel.c	2016-04-12 18:11:15.830625654 +0200
@@ -65,7 +65,7 @@
  * 4 bytes uint32    recipient channel
  * 4 bytes string    data
  */
-#define RECV_MAX_CHANNEL_DATA_LEN (RECV_MAX_PAYLOAD_LEN-(1+4+4))
+#define RECV_MAX_CHANNEL_DATA_LEN (RECV_MAX_PAYLOAD_LEN)
 
 /* Initialise all the channels */
 void chaninitialise(const struct ChanType *chantypes[]) {
