--- ltib/dist/lfs-5.1/expat/expat.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/expat/expat.spec	2015-10-12 10:46:10.738466000 +0200
@@ -14,6 +14,17 @@
 Patch2          : expat-1.95.8-man1dir.patch
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed expat for root file system requirements.
+Group           : Applications/Internet
+AutoReqProv     : no
+%Description rfs
+expat package contents restricted to just the files that are
+needed at run time on the target.
+
 
 %Description
 %{summary}
@@ -30,14 +41,22 @@
 
 %Install
 rm -rf $RPM_BUILD_ROOT
-make install DESTDIR=$RPM_BUILD_ROOT/%{pfx} install
-rm $RPM_BUILD_ROOT/%{pfx}/%{_prefix}/lib/*.la
+make install DESTDIR=$RPM_BUILD_ROOT%{pfx} install
+
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_prefix}/share/man, d;
+    \,^%{pfx}%{_prefix}/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
-
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
 
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
