--- ltib/dist/lfs-5.1/fontconfig/fontconfig.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/fontconfig/fontconfig.spec	2015-10-12 10:46:10.758465000 +0200
@@ -10,7 +10,18 @@
 Group           : System Environment/Libraries
 Source          : %{name}-%{version}.tar.gz
 BuildRoot       : %{_tmppath}/%{name}
-Prefix		    : %{pfx}
+Prefix		: %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed fontconfig for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+fontconfig package contents restricted to just the files that
+are needed at run time on the target.
+
 
 %Description
 %{summary}
@@ -24,13 +35,16 @@
 ac_cv_prog_CC_FOR_BUILD="${BUILDCC}" \
 ./configure --prefix=%{_prefix} --host=$CFGHOST --build=%{_build} \
             --with-arch=$GNUTARCH --sysconfdir=%{_sysconfdir} \
-            --with-freetype-config="$DEV_IMAGE/usr/bin/freetype-config --prefix=$DEV_IMAGE/usr"
+            --with-freetype-config="$DEV_IMAGE%{_prefix}/bin/freetype-config --prefix=$DEV_IMAGE%{_prefix}"
 make
 
 %Install
 rm -rf $RPM_BUILD_ROOT
-make install DESTDIR=$RPM_BUILD_ROOT/%{pfx} 
-rm -rf $RPM_BUILD_ROOT/%{pfx}/%{_prefix}/lib/*.la
+make install DESTDIR=$RPM_BUILD_ROOT%{pfx} 
+
+rm -rf $RPM_BUILD_ROOT%{pfx}%{_prefix}/lib/*.la
+rm -rf $RPM_BUILD_ROOT%{pfx}%{_prefix}/share/doc
+rm -rf $RPM_BUILD_ROOT%{pfx}%{_prefix}/lib/pkgconfig
 
 perl -pi -e 's,^</fontconfig>,
 <dir>%{_prefix}/X11R6/lib/X11/fonts/TTF</dir>
@@ -38,13 +52,23 @@
 <dir>%{_prefix}/X11R6/lib/X11/fonts/truetype</dir>
 </fontconfig>
 ,;
-        ' $RPM_BUILD_ROOT/%{pfx}/%{_sysconfdir}/fonts/local.conf
+        ' $RPM_BUILD_ROOT%{pfx}%{_sysconfdir}/fonts/local.conf
+
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_prefix}/share/man, d;
+    \,^%{pfx}%{_prefix}/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
 
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
