--- ltib/dist/lfs-5.1/glib2/glib2.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/glib2/glib2.spec	2015-10-12 10:46:10.770465000 +0200
@@ -1,24 +1,36 @@
 %define pfx /opt/freescale/rootfs/%{_target_cpu}
+%define archname glib
 
 Summary         : A library of functions used by GDK, GTK+, and many applications
 Name            : glib2
-Version         : 2.22.1
+Version         : 2.18.1
 Release         : 1
 License         : LGPL
 Vendor          : Freescale
 Packager        : Stuart Hughes/Kurt Mahan
 Group           : System Environment/Libraries
-Source          : glib-%{version}.tar.bz2
-Patch1          : glib2-2.12.11-relink.patch
+Source          : %{archname}-%{version}.tar.bz2
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed glib2 for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+
+
+%Description rfs
+glib2 package contents restricted to just the files that are
+needed at run time on the target.
+
 
 %Description
 %{summary}
 
 %Prep
-%setup -n glib-%{version}
-#%patch1 -p1
+%setup -n %{archname}-%{version}
 
 %Build
 # prevent configure from trying to compile and
@@ -27,21 +39,31 @@
 glib_cv_uscore=no \
 ac_cv_func_posix_getpwuid_r=yes \
 ac_cv_func_posix_getgrgid_r=yes \
-glib_cv_long_long_format=ll \
-ac_cv_func_printf_unix98=yes \
-ac_cv_func_vsnprintf_c99=yes \
 ./configure --prefix=%{_prefix} --host=$CFGHOST --build=%{_build}
 make
 
 %Install
 rm -rf $RPM_BUILD_ROOT
-make install DESTDIR=$RPM_BUILD_ROOT/%{pfx}
-find $RPM_BUILD_ROOT/%{pfx}/%{_prefix}/lib -name "*.la" | xargs rm -f
+make install DESTDIR=$RPM_BUILD_ROOT%{pfx}
+
+cd %{_builddir}/%{archname}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_datadir}/gtk-doc, d;
+    \,^%{pfx}%{_datadir}/locale, d;
+    \,^%{pfx}%{_mandir}, d;
+    \,^%{pfx}%{_libdir}/pkgconfig, d;
+    \,^%{pfx}%{_libdir}/glib-2.0, d;
+    \,^%{pfx}%{_includedir}, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
 
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
