diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/linux.bd imx-bootlets-src-11.01.00.modified/linux.bd
--- imx-bootlets-src-11.01.00/linux.bd	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/linux.bd	2012-03-19 15:13:36.355103796 +0100
@@ -7,7 +7,7 @@
 	power_prep="./power_prep/power_prep";
 	sdram_prep="./boot_prep/boot_prep";
 	linux_prep="./linux_prep/output-target/linux_prep";
-	zImage = "./zImage";
+	zImage="/imx_mect/trunk/imx28/ltib/rootfs/boot/zImage";
 }
 
 section (0) {
diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/linux_ivt.bd imx-bootlets-src-11.01.00.modified/linux_ivt.bd
--- imx-bootlets-src-11.01.00/linux_ivt.bd	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/linux_ivt.bd	2012-03-19 15:13:36.367103799 +0100
@@ -7,7 +7,7 @@
 	power_prep="./power_prep/power_prep";
 	sdram_prep="./boot_prep/boot_prep";
 	linux_prep="./linux_prep/output-target/linux_prep";
-	zImage="./zImage";
+	zImage="/imx_mect/trunk/imx28/ltib/rootfs/boot/zImage";
 }
 
 section (0) {
Binary files imx-bootlets-src-11.01.00/linux_prep/board/iMX28_EVK.o and imx-bootlets-src-11.01.00.modified/linux_prep/board/iMX28_EVK.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/libddi.a and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/libddi.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/libddi_power.a and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/libddi_power.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/src/ddi_power.o and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/src/ddi_power.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/src/ddi_power_5v_detect.o and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/src/ddi_power_5v_detect.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/src/ddi_power_battery.o and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/src/ddi_power_battery.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/src/ddi_power_handoff.o and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/src/ddi_power_handoff.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/drivers/power/src/ddi_power_init.o and imx-bootlets-src-11.01.00.modified/mach-mx28/drivers/power/src/ddi_power_init.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/core/libhw_core.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/core/libhw_core.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/core/src/hw_core_asm_interrupt.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/core/src/hw_core_asm_interrupt.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/digctl/libhw_digctl.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/digctl/libhw_digctl.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/digctl/src/hw_digctl.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/digctl/src/hw_digctl.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/icoll/libhw_icoll.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/icoll/libhw_icoll.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/icoll/src/hw_icoll_init.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/icoll/src/hw_icoll_init.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/icoll/src/hw_icoll_set.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/icoll/src/hw_icoll_set.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/libhw.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/libhw.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/libhw_lradc.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/libhw_lradc.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc_init.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc_init.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc_ladder.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc_ladder.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc_setup.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc_setup.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc_temperature_sensor.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc_temperature_sensor.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/lradc/src/hw_lradc_touch_screen.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/lradc/src/hw_lradc_touch_screen.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/libhw_power.a and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/libhw_power.a differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_5v_detect.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_5v_detect.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_battery.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_battery.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_conversions.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_conversions.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_fiq.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_fiq.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_init.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_init.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_interrupts.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_interrupts.o differ
Binary files imx-bootlets-src-11.01.00/mach-mx28/hw/power/src/hw_power_registers.o and imx-bootlets-src-11.01.00.modified/mach-mx28/hw/power/src/hw_power_registers.o differ
Binary files imx-bootlets-src-11.01.00/power_prep/debug.o and imx-bootlets-src-11.01.00.modified/power_prep/debug.o differ
diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/power_prep/power_prep.c imx-bootlets-src-11.01.00.modified/power_prep/power_prep.c
--- imx-bootlets-src-11.01.00/power_prep/power_prep.c	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/power_prep/power_prep.c	2012-03-19 15:19:17.047103797 +0100
@@ -144,7 +144,8 @@
 * of the recommended operating voltage range from the characteristics and
 * specifications section of the mx23/mx28 reference manual
 */
-/* #define NO_BATTERY_VOLTAGE_SOURCE */
+#define NO_BATTERY_VOLTAGE_SOURCE
+//#define CONFIG_MECT_CUSTOMIZATION
 
 #if 0
 #define RUN_BATTERY_TEST
@@ -331,9 +332,10 @@
 		Battery powered operation disabled.\r\n");
 
 	/* disable automatic battery voltage measurements */
+#ifndef CONFIG_MECT_CUSTOMIZATION
 	BF_CLR(LRADC_CONVERSION, AUTOMATIC);
 	BF_WR(POWER_BATTMONITOR, BATT_VAL,0);
-
+#endif
 	iReturnValue = PowerPrep_5vBoot();
 
 #ifndef MXS_VBUS_CURRENT_DRAW
diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/uboot.bd imx-bootlets-src-11.01.00.modified/uboot.bd
--- imx-bootlets-src-11.01.00/uboot.bd	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/uboot.bd	2012-03-19 15:13:37.763103796 +0100
@@ -3,7 +3,7 @@
 sources {
 	power_prep="./power_prep/power_prep";
 	sdram_prep="./boot_prep/boot_prep";
-	u_boot = "./u-boot";
+	u_boot="/imx_mect/trunk/imx28/ltib/rootfs/boot/u-boot";
 }
 
 section (0) {
diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/uboot_ivt.bd imx-bootlets-src-11.01.00.modified/uboot_ivt.bd
--- imx-bootlets-src-11.01.00/uboot_ivt.bd	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/uboot_ivt.bd	2012-03-19 15:13:37.767103797 +0100
@@ -3,7 +3,7 @@
 sources {
 	power_prep="./power_prep/power_prep";
 	sdram_prep="./boot_prep/boot_prep";
-	u_boot="./u-boot";
+	u_boot="/imx_mect/trunk/imx28/ltib/rootfs/boot/u-boot";
 }
 
 section (0) {
