diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/linux_prep/Makefile imx-bootlets-src-11.01.00.modified/linux_prep/Makefile
--- imx-bootlets-src-11.01.00/linux_prep/Makefile	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/linux_prep/Makefile	2014-12-05 18:02:11.000000000 +0100
@@ -1,3 +1,6 @@
+MYLIBGCCDIR	= /home/imx28/CodeSourcery/Sourcery_G++_Lite/arm-none-linux-gnueabi/lib
+MYYLIBGCCDIR	= /home/imx28/CodeSourcery/Sourcery_G++_Lite/arm-none-linux-gnueabi/libc
+
 CURRENTDIR	= .
 SRCDIR		= $(CURRENTDIR)
 OUTPUTDIR	= $(CURRENTDIR)/output-target
@@ -25,6 +28,7 @@
 AS	= $(CROSS_COMPILE)as
 CC	= $(CROSS_COMPILE)gcc
 LD	= $(CROSS_COMPILE)ld
+#LD	= $(CROSS_COMPILE)gcc
 CPP	= $(CROSS_COMPILE)cpp
 STRIP	= $(CROSS_COMPILE)strip
 OBJCOPY	= $(CROSS_COMPILE)objcopy
@@ -32,7 +36,8 @@
 
 LIBGCCDIR = $(dir $(shell $(CC) -print-libgcc-file-name))
 CFLAGS 	= -Wall -I$(INCLUDEDIR) -nostdinc -fno-builtin -O
-LDFLAGS = -static -nostdlib -T $(BOOT_LAYOUT_OUT) -L$(LIBGCCDIR) -lgcc
+LDFLAGS = -static -nostdlib -T $(BOOT_LAYOUT_OUT) -L$(LIBGCCDIR) -L$(MYLIBGCCDIR) -L$(MYYLIBGCCDIR) -lgcc 
+#LDFLAGS = -static -T $(BOOT_LAYOUT_OUT) -L$(LIBGCCDIR) -L$(MYLIBGCCDIR) -L$(MYYLIBGCCDIR) -lgcc,-lc,-lgcc,-lc,lsupc++ 
 
 BOARD ?= stmp378x_dev
 
diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/linux_prep/hw/lradc-buttons.c imx-bootlets-src-11.01.00.modified/linux_prep/hw/lradc-buttons.c
--- imx-bootlets-src-11.01.00/linux_prep/hw/lradc-buttons.c	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/linux_prep/hw/lradc-buttons.c	2014-12-05 17:49:30.000000000 +0100
@@ -22,6 +22,10 @@
 #define TARGET_VDDIO_LRADC_VAL	3350
 #define LRADC_NOISE_MARGIN	100
 
+ void __aeabi_unwind_cpp_pr1(void)
+ {
+ };
+
 static u32 lradc_get_ch_value (int channel);
 
 static u32 lradc_decode_button (int raw_button)
