diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/Makefile imx-bootlets-src-11.01.00.modified/Makefile
--- imx-bootlets-src-11.01.00/Makefile	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/Makefile	2012-05-25 18:08:47.768175919 +0200
@@ -2,7 +2,9 @@
 MEM_TYPE ?= MEM_DDR1
 export MEM_TYPE
 
-DFT_IMAGE=$(DEV_IMAGE)/boot/zImage
+DFT_IMAGE=$(DEV_IMAGE)/boot$(CURR_KERN_VER)/zImage
+	
+#DFT_IMAGE=$(DEV_IMAGE)/boot_xeno/zImage
 DFT_UBOOT=$(DEV_IMAGE)/boot/u-boot
 
 BOARD ?= stmp378x_dev
@@ -24,6 +26,7 @@
 
 gen_bootstream: linux_prep boot_prep power_prep linux.bd uboot.bd linux.bd uboot.bd updater.bd
 	@echo "generating linux kernel boot stream image"
+	echo $DFT_IMAGE
 ifeq "$(DFT_IMAGE)" "$(wildcard $(DFT_IMAGE))"
 	@echo "by using the rootfs/boot/zImage"
 	sed -i 's,[^ *]zImage.*;,\tzImage="$(DFT_IMAGE)";,' linux.bd

