diff --exclude CVS --exclude .git -uNr imx-bootlets-src-11.01.00/boot_prep/init-mx28.c imx-bootlets-src-11.01.00.modified/boot_prep/init-mx28.c
--- imx-bootlets-src-11.01.00/boot_prep/init-mx28.c	2010-11-04 08:35:38.000000000 +0100
+++ imx-bootlets-src-11.01.00.modified/boot_prep/init-mx28.c	2014-10-07 16:46:58.003189061 +0200
@@ -1025,6 +1025,14 @@
 	div = (~div);
 	HW_CLKCTRL_FRAC0_CLR(BF_CLKCTRL_FRAC0_EMIFRAC(div));
 }
+
+void set_gpmi_frac(unsigned int div)
+{
+	HW_CLKCTRL_FRAC1_SET(BM_CLKCTRL_FRAC1_GPMIFRAC);
+	div = (~div);
+	HW_CLKCTRL_FRAC1_CLR(BF_CLKCTRL_FRAC1_GPMIFRAC(div));
+}
+
 void init_clock()
 {
 	HW_CLKCTRL_FRAC0_SET(BM_CLKCTRL_FRAC0_CLKGATEEMI);
@@ -1042,7 +1050,23 @@
 	HW_CLKCTRL_CLKSEQ_CLR(BM_CLKCTRL_CLKSEQ_BYPASS_EMI);
 
 	printf("FRAC 0x%x\r\n" , HW_CLKCTRL_FRAC0_RD());
+//Init clock for GPMI NAND FLASH to be driven by ref_gpmi and not by ref_xtal
+
+	HW_CLKCTRL_FRAC1_SET(BM_CLKCTRL_FRAC1_CLKGATEGPMI);
+
+
+	set_gpmi_frac(18);
+
+	HW_CLKCTRL_FRAC1_CLR(BM_CLKCTRL_FRAC1_CLKGATEGPMI);
+	delay(11000);
+
+	HW_CLKCTRL_GPMI_WR(
+			  BF_CLKCTRL_GPMI_DIV(3)
+			 );
 
+	/*choose ref_gpmi*/
+	HW_CLKCTRL_CLKSEQ_CLR(BM_CLKCTRL_CLKSEQ_BYPASS_GPMI);
+	printf("FRAC1 0x%x\r\n" , HW_CLKCTRL_FRAC1_RD());
 }
 
 void disable_emi_padkeepers(void)
