--- ltib/dist/lfs-5.1/kernel/kernel-common.tmpl	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/kernel/kernel-common.tmpl	2015-10-27 18:55:27.932880003 +0100
@@ -182,10 +182,10 @@
 #
 # build the kernel and optionally the modules
 #
-make $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" $SYSCFG_KTARG
+make -j`grep -c ^processor /proc/cpuinfo` $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" $SYSCFG_KTARG
 if grep -q '^CONFIG_MODULES=' $KBOUT/.config
 then
-    make $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" modules
+    make -j`grep -c ^processor /proc/cpuinfo` $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" modules
 fi
 
 # cscope index
@@ -333,6 +333,14 @@
     fi
 fi
 
+cd %{_builddir}/%{pkg_name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}/boot, d;
+    \,^%{pfx}%{_prefix}/src, d;
+    \,^%{pfx}/%{_lib}/modules/%{version}-[^/]*/kernel/drivers/usb/serial/usbserial.ko, d;
+' all_files.lst > rfs_files.lst
+cd -
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
@@ -341,6 +349,13 @@
     rm -f $RPM_BUILD_DIR/linux
 fi
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
+
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
+
+%Files dev-usb_serial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/usbserial.ko
