diff --exclude CVS --exclude .git -uNr linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c linux-2.6.35.3.modified/arch/arm/mach-mx28/mx28evk_pins.c
--- linux-2.6.35.3/arch/arm/mach-mx28/mx28evk_pins.c	2012-03-29 18:55:35.279291940 +0200
+++ linux-2.6.35.3.modified/arch/arm/mach-mx28/mx28evk_pins.c	2012-03-29 18:10:28.423073996 +0200
@@ -1384,10 +1384,8 @@
 
 #if defined CONFIG_MECT_VAL01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 1);
-#else
-	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
-#endif
-#if defined CONFIG_MECT_ELE01
+
+#elif CONFIG_MECT_ELE01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
 #else
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
@@ -1414,11 +1412,10 @@
 	int i;
 #if defined CONFIG_MECT_VAL01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
-#else
-	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 1);
-#endif
-#if defined CONFIG_MECT_ELE01
+
+#elif defined CONFIG_MECT_ELE01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 1);
+
 #else
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 1);
 #endif
@@ -1443,11 +1440,10 @@
 
 #if defined CONFIG_MECT_VAL01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 1);
-#else
-	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
-#endif
-#if defined CONFIG_MECT_ELE01
+
+#elif defined CONFIG_MECT_ELE01
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
+
 #else
 	gpio_direction_output(MXS_PIN_TO_GPIO(PINID_SSP1_DATA3), 0);
 #endif

