--- ltib/dist/lfs-5.1/kernel/kernel-common.tmpl	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/kernel/kernel-common.tmpl	2015-10-29 17:57:23.648458710 +0100
@@ -182,10 +182,10 @@
 #
 # build the kernel and optionally the modules
 #
-make $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" $SYSCFG_KTARG
+make -j`grep -c ^processor /proc/cpuinfo` $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" $SYSCFG_KTARG
 if grep -q '^CONFIG_MODULES=' $KBOUT/.config
 then
-    make $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" modules
+    make -j`grep -c ^processor /proc/cpuinfo` $LOCV ARCH=$LINTARCH CROSS_COMPILE= HOSTCC="$BUILDCC" modules
 fi
 
 # cscope index
@@ -333,6 +333,14 @@
     fi
 fi
 
+cd %{_builddir}/%{pkg_name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}/boot, d;
+    \,^%{pfx}%{_prefix}/src, d;
+    \,^%{pfx}/%{_lib}/modules/%{version}-[^/]*/kernel/drivers/usb/serial/usbserial.ko, d;
+' all_files.lst > rfs_files.lst
+cd -
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
@@ -341,6 +349,200 @@
     rm -f $RPM_BUILD_DIR/linux
 fi
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
+
+%Files dev-usbserial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/usbserial.ko
+
+%Files dev-usbserial-aircable
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/aircable.ko
+
+%Files dev-usbserial-ark3116
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ark3116.ko
+
+%Files dev-usbserial-belkin_sa
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/belkin_sa.ko
+
+%Files dev-usbserial-ch341
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ch341.ko
+
+%Files dev-usbserial-cp210x
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/cp210x.ko
+
+%Files dev-usbserial-cyberjack
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/cyberjack.ko
+
+%Files dev-usbserial-cypress_m8
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/cypress_m8.ko
+
+%Files dev-usbserial-digi_acceleport
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/digi_acceleport.ko
+
+%Files dev-usbserial-empeg
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/empeg.ko
+
+%Files dev-usbserial-ftdi_sio
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ftdi_sio.ko
+
+%Files dev-usbserial-funsoft
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/funsoft.ko
+
+%Files dev-usbserial-garmin_gps
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/garmin_gps.ko
+
+%Files dev-usbserial-hp4x
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/hp4x.ko
+
+%Files dev-usbserial-io_edgeport
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/io_edgeport.ko
+
+%Files dev-usbserial-io_ti
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/io_ti.ko
+
+%Files dev-usbserial-ipaq
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ipaq.ko
+
+%Files dev-usbserial-ipw
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ipw.ko
+
+%Files dev-usbserial-ir
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ir-usb.ko
+
+%Files dev-usbserial-iuu_phoenix
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/iuu_phoenix.ko
+
+%Files dev-usbserial-keyspan
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/keyspan.ko
+
+%Files dev-usbserial-keyspan_pda
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/keyspan_pda.ko
+
+%Files dev-usbserial-kl5kusb105
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/kl5kusb105.ko
+
+%Files dev-usbserial-kobil_sct
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/kobil_sct.ko
+
+%Files dev-usbserial-mct_u232
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/mct_u232.ko
+
+%Files dev-usbserial-mos7720
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/mos7720.ko
+
+%Files dev-usbserial-mos7840
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/mos7840.ko
+
+%Files dev-usbserial-moto_modem
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/moto_modem.ko
+
+%Files dev-usbserial-navman
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/navman.ko
+
+%Files dev-usbserial-omninet
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/omninet.ko
+
+%Files dev-usbserial-opticon
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/opticon.ko
+
+%Files dev-usbserial-option
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/option.ko
+
+%Files dev-usbserial-oti6858
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/oti6858.ko
+
+%Files dev-usbserial-pl2303
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/pl2303.ko
+
+%Files dev-usbserial-qcaux
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/qcaux.ko
+
+%Files dev-usbserial-qcserial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/qcserial.ko
+
+%Files dev-usbserial-safe_serial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/safe_serial.ko
+
+%Files dev-usbserial-siemens_mpi
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/siemens_mpi.ko
+
+%Files dev-usbserial-sierra
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/sierra.ko
+
+%Files dev-usbserial-spcp8x5
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/spcp8x5.ko
+
+%Files dev-usbserial-symbolserial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/symbolserial.ko
+
+%Files dev-usbserial-ti_3410_5052
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/ti_usb_3410_5052.ko
+
+%Files dev-usbserial-debug
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/usb_debug.ko
+
+%Files dev-usbserial-wwan
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/usb_wwan.ko
+
+%Files dev-usbserial-visor
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/visor.ko
+
+%Files dev-usbserial-vivopay-serial
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/vivopay-serial.ko
+
+%Files dev-usbserial-whiteheat
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/whiteheat.ko
+
+%Files dev-usbserial-zio
+%defattr(-,root,root)
+%{pfx}/%{_lib}/modules/%{version}-*/kernel/drivers/usb/serial/zio.ko
