diff --exclude CVS --exclude .git -uNr linux-2.6.35.3/arch/arm/mach-mx28/device.c linux-2.6.35.3.modified/arch/arm/mach-mx28/device.c
--- linux-2.6.35.3/arch/arm/mach-mx28/device.c	2014-12-06 10:18:12.383356962 +0100
+++ linux-2.6.35.3.modified/arch/arm/mach-mx28/device.c	2014-12-06 10:17:01.087728521 +0100
@@ -449,7 +449,7 @@
 #endif
 
 #if defined(CONFIG_MMC_MXS) || defined(CONFIG_MMC_MXS_MODULE)
-#if defined(CONFIG_MACH_MX28EVK) && !defined(CONFIG_MECT_ATN01)
+#if defined(CONFIG_MACH_MX28EVK) && !(defined(CONFIG_MECT_ATN01) || defined(CONFIG_MECT_TPAC1007_480x272))
 #define MMC0_POWER	MXS_PIN_TO_GPIO(PINID_PWM3)
 #define MMC1_POWER	MXS_PIN_TO_GPIO(PINID_PWM4)
 #define MMC0_WP		MXS_PIN_TO_GPIO(PINID_SSP1_SCK)
