--- a/drivers/char/mxs-buzzer.c	2017-11-20 12:42:20.824192440 +0000
+++ b/drivers/char/mxs-buzzer.c	2017-11-20 13:36:52.242839659 +0000
@@ -42,11 +42,10 @@
 #define SCALE_Q		0  
 #define SCALE_M		10
 
+#define BUZZER_CLOCK_CYCLE_IN_PERIOD 3750 // @ (24MHz/64=2.667us)=10ms
 #if defined CONFIG_MECT_TPAC1005_480x272
-#define BUZZER_CLOCK_CYCLE_IN_PERIOD 	749 //500Hz //374 1kHz //81 corresponds to a frequency for pwm = 3.5 Khz
 struct buzzer_mesg settings;
 #else
-#define BUZZER_CLOCK_CYCLE_IN_PERIOD 	599
 #endif
 
 static const struct file_operations buzzer_fops ={
@@ -145,7 +144,7 @@
 #if defined CONFIG_MECT_TPAC1005_480x272
 	__raw_writel(BF_PWM_ACTIVEn_INACTIVE(duty_cycle) |
 #else	
-	__raw_writel(BF_PWM_ACTIVEn_INACTIVE(50) |
+	__raw_writel(BF_PWM_ACTIVEn_INACTIVE(BUZZER_CLOCK_CYCLE_IN_PERIOD) |
 #endif
 		     BF_PWM_ACTIVEn_ACTIVE(0),
 		     REGS_PWM_BASE + HW_PWM_ACTIVEn(3));
