--- ltib/dist/lfs-5.1/libsocketcan/libsocketcan.spec	1970-01-01 01:00:00.000000000 +0100
+++ ltib.new/dist/lfs-5.1/libsocketcan/libsocketcan.spec	2015-10-12 10:46:10.502469000 +0200
@@ -0,0 +1,65 @@
+%define pfx /opt/freescale/rootfs/%{_target_cpu}
+
+Summary         : Library to allow control of basic functions in socketcan from userspace
+Name            : libsocketcan
+Version         : 0.0.8
+Release         : 0
+License         : GNU LGPLv2.1
+Vendor          : Freescale
+Packager        : Rogerio Pimentel
+Group           : System Environment/Libraries
+Source          : %{name}-%{version}.tar.gz
+URL             : http://www.pengutronix.de/software/libsocketcan/download/
+Patch1          : libsocketcan-0.0.8-1317716454.patch
+BuildRoot       : %{_tmppath}/%{name}
+Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed libsocketcan for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+libsocketcan package contents restricted to just the files
+that are needed at run time on the target.
+
+
+%Description
+%{summary}
+
+%Prep
+%setup
+%patch1 -p1
+
+%Build
+
+
+./configure --prefix=%{_prefix} --host=$CFGHOST --build=%{_build}
+#./configure --prefix=%{_prefix} --host=$CFGHOST --build=%{_build} CC=/opt/freescale/usr/local/gcc-4.4.4-glibc-2.11.1-multilib-1.0/arm-fsl-linux-gnueabi/bin/arm-fsl-linux-gnueabi-gcc
+
+make
+
+%Install
+rm -rf $RPM_BUILD_ROOT
+make install DESTDIR=$RPM_BUILD_ROOT/%{pfx}
+
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_libdir}/pkgconfig, d;
+    \,^%{pfx}%{_datadir}/doc, d;
+    \,^%{pfx}%{_prefix}/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
+
+%Clean
+rm -rf $RPM_BUILD_ROOT
+
+
+%Files -f all_files.lst
+%defattr(-,root,root)
+
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
