--- ltib/dist/lfs-5.1/libtermcap/libtermcap.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/libtermcap/libtermcap.spec	2015-10-12 10:46:10.778465000 +0200
@@ -1,4 +1,5 @@
 %define pfx /opt/freescale/rootfs/%{_target_cpu}
+%define archname termcap
 
 Summary         : A basic system library for accessing the termcap database.
 Name            : libtermcap
@@ -8,21 +9,32 @@
 Vendor          : Freescale Inc
 Packager        : Steve Papacharalambous
 Group           : System Environment/Libraries
-Source          : termcap-2.0.8.tar.bz2
-Patch0: termcap-2.0.8-shared.patch
-Patch1: termcap-2.0.8-setuid.patch
-Patch2: termcap-2.0.8-instnoroot.patch
-Patch3: termcap-2.0.8-compat21.patch
-Patch4: termcap-2.0.8-xref.patch
-Patch5: termcap-2.0.8-fix-tc.patch
-Patch6: termcap-2.0.8-ignore-p.patch
-Patch7: termcap-buffer.patch
-Patch8: termcap-2.0.8-bufsize.patch
-Patch9: termcap-2.0.8-colon.patch
-Patch10: libtermcap-aaargh.patch
-Patch11: termcap-2.0.8-glibc22.patch
+Source          : %{archname}-%{version}.tar.bz2
+Patch0          : %{archname}-%{version}-shared.patch
+Patch1          : %{archname}-%{version}-setuid.patch
+Patch2          : %{archname}-%{version}-instnoroot.patch
+Patch3          : %{archname}-%{version}-compat21.patch
+Patch4          : %{archname}-%{version}-xref.patch
+Patch5          : %{archname}-%{version}-fix-tc.patch
+Patch6          : %{archname}-%{version}-ignore-p.patch
+Patch7          : %{archname}-buffer.patch
+Patch8          : %{archname}-%{version}-bufsize.patch
+Patch9          : %{archname}-%{version}-colon.patch
+Patch10         : libtermcap-aaargh.patch
+Patch11         : %{archname}-%{version}-glibc22.patch
 BuildRoot       : %{_tmppath}/%{name}
 Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed libtermcap for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+libtermcap package contents restricted to just the files that
+are needed at run time on the target.
+
 
 %Description
 The libtermcap package contains a basic system library needed to
@@ -31,7 +43,7 @@
 displays in a terminal-independent manner.
 
 %Prep
-%setup -q -n termcap-2.0.8
+%setup -q -n %{archname}-%{version}
 %patch0 -p1 
 %patch1 -p1
 %patch2 -p1
@@ -61,7 +73,7 @@
 if [ "$GNUTARCH" = m68knommu ]
 then
 ( cp libtermcap.a ${RPM_BUILD_ROOT}/%{pfx}/lib/
-  cp termcap.h ${RPM_BUILD_ROOT}/%{pfx}/usr/include/
+  cp termcap.h ${RPM_BUILD_ROOT}/%{pfx}%{_prefix}/include/
   cp termcap.info* ${RPM_BUILD_ROOT}/%{pfx}/%{_infodir}
   cd ${RPM_BUILD_ROOT}/%{pfx}
   cd ./%{_infodir}
@@ -71,7 +83,7 @@
 
 else
 export PATH=/sbin:$PATH
-make prefix=${RPM_BUILD_ROOT}/%{pfx}/usr install
+make prefix=${RPM_BUILD_ROOT}/%{pfx}%{_prefix} install
 
 install -c -m644 termcap.src ${RPM_BUILD_ROOT}/%{pfx}/etc/termcap
 cp termcap.info* ${RPM_BUILD_ROOT}/%{pfx}/%{_infodir}
@@ -79,24 +91,33 @@
 ( cd ${RPM_BUILD_ROOT}/%{pfx}
   rm -f ./etc/termcap
   mv ./usr/lib/libtermcap.so* ./lib
-  ln -sf libtermcap.so.2.0.8 ./lib/libtermcap.so.2
+  ln -sf libtermcap.so.%{version} ./lib/libtermcap.so.2
   ln -sf libtermcap.so.2 ./lib/libtermcap.so
   cd usr/lib
-  ln -sf ../../lib/libtermcap.so.2.0.8 libtermcap.so
+  ln -sf ../../lib/libtermcap.so.%{version} libtermcap.so
   cd ../..
-  strip -R .comments --strip-unneeded ./lib/libtermcap.so.2.0.8
+  strip -R .comments --strip-unneeded ./lib/libtermcap.so.%{version}
   cd ./%{_infodir}
   gzip -9nf ./termcap.info*
   chmod 644 ./termcap.info*
 )
 fi
 
+cd %{_builddir}/%{archname}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_prefix}/info, d;
+    \,^%{pfx}%{_prefix}/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
+
 %Clean
 rm -rf ${RPM_BUILD_ROOT}
-rm -rf ${RPM_BUILD_DIR}/termcap-%{version}
+rm -rf ${RPM_BUILD_DIR}/%{archname}-%{version}
 
-%Files
+%Files -f all_files.lst
 %defattr(-,root,root)
-%{pfx}/*
 
 
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
