--- ltib.orig/ltib	2015-10-29 22:19:16.449630271 +0100
+++ ltib/ltib	2015-11-09 13:45:30.026001862 +0100
@@ -265,7 +265,7 @@
 
 # package build dependencies
 # rebuild all packages in the list (rhs) if the key (lhs) has been installed
-$build_deps = { PKG_KERNEL => [ qw/PKG_MODEPS/ ],
+$build_deps = { PKG_KERNEL => [ qw/PKG_BOOT_STREAM PKG_MODEPS PKG_LAST_KERNEL_BUILD/ ],
                 PKG_SKELL  => [ qw/PKG_SYSCONFIG/ ],
                 PKG_MERGE  => [ qw/PKG_MODEPS/ ],
                 PKG_TSLIB  => [ qw/PKG_XORG_SERVER PKG_ENLIGHTENMENT
@@ -1611,7 +1611,7 @@
 
 sub save_binary_rpms
 {
-    die("pcf  not defined") unless $pcf;
+    die("pcf not defined") unless $pcf;
 
     my $newarch = g2larch($pcf->{LINTARCH}) or die;
     my $cfn     = '.config';
