--- ltib.original/bin/Ltibutils.pm	2013-01-09 04:58:38.000000000 +0100
+++ ltib/bin/Ltibutils.pm	2015-08-27 11:48:26.933631292 +0200
@@ -359,7 +359,7 @@
     return scalar @{$cf->{pp_list}} if $pp_list_tested;
     $pp_list_tested = 1;
 
-    return unless defined @{$cf->{pp_list}};
+    return unless @{$cf->{pp_list}};
     return 1 if $cf->{bypass_network_tests};
     return 1 unless $have_request_common && $have_useragent;
 
@@ -584,6 +584,7 @@
                     },
     zlib         => sub { my @f = (glob('/usr/lib/libz.so*'),
                                    glob('/lib/libz.so*'),
+                                   glob('/lib/i386-linux-gnu/libz.so*'),
                                    glob('/lib64/libz.so*'),
                                    glob('/usr/lib/i386-linux-gnu/libz.so*'),
                                    glob('/usr/lib32/libz.so*'),
