diff -aurN '--exclude=*-orig.spec' '--exclude=*.bak' '--exclude=*.swp' ltib.reference/dist/lfs-5.1/mect_apps/mect_apps.spec ltib/dist/lfs-5.1/mect_apps/mect_apps.spec
--- ltib.reference/dist/lfs-5.1/mect_apps/mect_apps.spec
+++ ltib/dist/lfs-5.1/mect_apps/mect_apps.spec
@@ -833,10 +833,10 @@
 	install -m644 -D config/system.ini     $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/etc/sysconfig/system.ini
 	install -m644 -D config/trend1.csv     $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/customtrend/trend1.csv
 
-	install -m644 -D recipes/AnOut/up_and_down.csv  $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipes/AnOut/up_and_down.csv
-	install -m644 -D recipes/AnOut/sine_wave.csv    $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipes/AnOut/sine_wave.csv
-	install -m644 -D recipes/DigOut/from_1_to_4.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipes/DigOut/from_1_to_4.csv
-	install -m644 -D recipes/DigOut/from_1_to_8.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipes/DigOut/from_1_to_8.csv
+	install -m644 -D recipes/AnOut/up_and_down.csv  $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipe/AnOut/up_and_down.csv
+	install -m644 -D recipes/AnOut/sine_wave.csv    $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipe/AnOut/sine_wave.csv
+	install -m644 -D recipes/DigOut/from_1_to_4.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipe/DigOut/from_1_to_4.csv
+	install -m644 -D recipes/DigOut/from_1_to_8.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_03/local/data/recipe/DigOut/from_1_to_8.csv
 
 	cd -
 fi
@@ -883,10 +883,10 @@
 	install -m644 -D config/system.ini     $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/etc/sysconfig/system.ini
 	install -m644 -D config/trend1.csv     $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/customtrend/trend1.csv
 
-	install -m644 -D recipes/AnOut/up_and_down.csv  $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipes/AnOut/up_and_down.csv
-	install -m644 -D recipes/AnOut/sine_wave.csv    $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipes/AnOut/sine_wave.csv
-	install -m644 -D recipes/DigOut/from_1_to_4.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipes/DigOut/from_1_to_4.csv
-	install -m644 -D recipes/DigOut/from_1_to_8.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipes/DigOut/from_1_to_8.csv
+	install -m644 -D recipes/AnOut/up_and_down.csv  $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipe/AnOut/up_and_down.csv
+	install -m644 -D recipes/AnOut/sine_wave.csv    $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipe/AnOut/sine_wave.csv
+	install -m644 -D recipes/DigOut/from_1_to_4.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipe/DigOut/from_1_to_4.csv
+	install -m644 -D recipes/DigOut/from_1_to_8.csv $RPM_BUILD_ROOT%{pfx}/TPAC1007_04_AC/local/data/recipe/DigOut/from_1_to_8.csv
 
 	cd -
 fi
@@ -1130,10 +1130,10 @@
 %{pfx}/TPAC1007_03/local/etc/sysconfig/img/splash.png
 %{pfx}/TPAC1007_03/local/etc/sysconfig/system.ini
 %{pfx}/TPAC1007_03/local/data/customtrend/trend1.csv
-%{pfx}/TPAC1007_03/local/data/recipes/AnOut/up_and_down.csv
-%{pfx}/TPAC1007_03/local/data/recipes/AnOut/sine_wave.csv
-%{pfx}/TPAC1007_03/local/data/recipes/DigOut/from_1_to_4.csv
-%{pfx}/TPAC1007_03/local/data/recipes/DigOut/from_1_to_8.csv
+%{pfx}/TPAC1007_03/local/data/recipe/AnOut/up_and_down.csv
+%{pfx}/TPAC1007_03/local/data/recipe/AnOut/sine_wave.csv
+%{pfx}/TPAC1007_03/local/data/recipe/DigOut/from_1_to_4.csv
+%{pfx}/TPAC1007_03/local/data/recipe/DigOut/from_1_to_8.csv
 
 %files lfs-TPAC1007_04_AA
 %defattr(-,root,root)
@@ -1158,10 +1158,10 @@
 %{pfx}/TPAC1007_04_AC/local/etc/sysconfig/img/splash.png
 %{pfx}/TPAC1007_04_AC/local/etc/sysconfig/system.ini
 %{pfx}/TPAC1007_04_AC/local/data/customtrend/trend1.csv
-%{pfx}/TPAC1007_04_AC/local/data/recipes/AnOut/up_and_down.csv
-%{pfx}/TPAC1007_04_AC/local/data/recipes/AnOut/sine_wave.csv
-%{pfx}/TPAC1007_04_AC/local/data/recipes/DigOut/from_1_to_4.csv
-%{pfx}/TPAC1007_04_AC/local/data/recipes/DigOut/from_1_to_8.csv
+%{pfx}/TPAC1007_04_AC/local/data/recipe/AnOut/up_and_down.csv
+%{pfx}/TPAC1007_04_AC/local/data/recipe/AnOut/sine_wave.csv
+%{pfx}/TPAC1007_04_AC/local/data/recipe/DigOut/from_1_to_4.csv
+%{pfx}/TPAC1007_04_AC/local/data/recipe/DigOut/from_1_to_8.csv
 
 %files lfs-TPAC1007_LV
 %defattr(-,root,root)
