--- ltib/dist/lfs-5.1/pkgconfig/pkgconfig.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/pkgconfig/pkgconfig.spec	2015-10-12 10:46:10.434469000 +0200
@@ -24,7 +24,7 @@
 # don't add --build as glib will fail for a host build due to 
 # stupid incorrect cross compile detection
 ./configure --prefix=%{_prefix} --host=$CFGHOST
-make
+make -j`grep -c ^processor /proc/cpuinfo`
 
 %Install
 rm -rf $RPM_BUILD_ROOT
