diff --exclude CVS --exclude .git -uNr qt-everywhere-opensource-src-4.8.5/configure qt-everywhere-opensource-src-4.8.5.modified/configure
--- qt-everywhere-opensource-src-4.8.5/configure	2013-06-07 07:16:41.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.5.modified/configure	2014-02-25 14:23:01.843586886 +0100
@@ -690,7 +690,8 @@
 # initalize variables
 #-------------------------------------------------------------------------------
 
-SYSTEM_VARIABLES="RANLIB STRIP OBJDUMP LD CC CXX CFLAGS CXXFLAGS LDFLAGS"
+#SYSTEM_VARIABLES="RANLIB STRIP OBJDUMP LD CC CXX CFLAGS CXXFLAGS LDFLAGS"
+SYSTEM_VARIABLES="RANLIB STRIP OBJDUMP CC CXX CFLAGS CXXFLAGS LDFLAGS"
 for varname in $SYSTEM_VARIABLES; do
     qmakevarname="${varname}"
     # use LDFLAGS for autoconf compat, but qmake uses QMAKE_LFLAGS
diff --exclude CVS --exclude .git -uNr qt-everywhere-opensource-src-4.8.5/src/gui/embedded/qsoundqss_qws.cpp qt-everywhere-opensource-src-4.8.5.modified/src/gui/embedded/qsoundqss_qws.cpp
--- qt-everywhere-opensource-src-4.8.5/src/gui/embedded/qsoundqss_qws.cpp	2013-06-07 07:16:59.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.5.modified/src/gui/embedded/qsoundqss_qws.cpp	2014-02-25 14:23:50.035478805 +0100
@@ -69,7 +69,7 @@
 
 #include <qvfbhdr.h>
 
-extern int errno;
+//extern int errno;
 
 QT_BEGIN_NAMESPACE
 
