diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/autoexec skell-1.18.modified/etc/rc.d/init.d/autoexec
--- skell-1.18/etc/rc.d/init.d/autoexec	1970-01-01 01:00:00.000000000 +0100
+++ skell-1.18.modified/etc/rc.d/init.d/autoexec	2012-02-08 10:45:20.977530828 +0100
@@ -0,0 +1,59 @@
+#!/bin/sh
+
+start() {
+
+# ts variable
+export TSLIB_CONFFILE=/usr/etc/ts.conf
+export TSLIB_PLUGINDIR=/usr/lib/ts
+export TSLIB_TSDEVICE=/dev/input/ts0
+export TSLIB_CONSOLEDEVICE=none
+export TSLIB_CALIBFILE=/local/etc/sysconfig/pointercal
+
+# qt variable
+export QWS_MOUSE_PROTO=tslib:$TSLIB_TSDEVICE
+export QT_QWS_FONTDIR=/usr/lib/fonts
+export POINTERCAL_FILE=$TSLIB_CALIBFILE
+
+if [ ! -e $TSLIB_CALIBFILE ]; then
+	echo -n "Starting 'calibration'... "
+	/usr/bin/ts_calibrate
+	sleep 1
+	echo "Done."
+fi
+
+if [ -x /usr/bin/autoexec ]; then
+	echo -n "Starting 'autoexec'... "
+	start-stop-daemon  -S -b -q -x /usr/bin/autoexec
+	echo "Done."
+fi
+}
+
+stop() {
+echo -n "Shutting down 'autoexec'... "
+start-stop-daemon  -K -q -x /usr/bin/autoexec
+echo "Done."
+}
+
+restart() {
+stop
+sleep 1
+start
+}
+
+case "$1" in
+start)
+    start
+;;
+stop)
+    stop
+;;
+restart)
+    restart
+;;
+*)
+  echo $"Usage: $0 {start|stop|restart}"
+        exit 1
+    ;;
+    
+esac
+
