diff --exclude CVS --exclude .git -uNr skell-1.18/local/etc/sysconfig/net.conf skell-1.18.modified/local/etc/sysconfig/net.conf
--- skell-1.18/local/etc/sysconfig/net.conf	2016-03-23 15:45:26.057419730 +0100
+++ skell-1.18.modified/local/etc/sysconfig/net.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-#
-# set up the network interfaces
-#
-MAC0=00:11:22:22:11:00
-IPADDR0=192.168.5.211
-NETMASK0=255.255.255.0
-GATEWAY0=192.168.5.10
-NAMESERVER01=1.0.0.0
-NAMESERVER02=1.0.0.0
-MAC1=
-IPADDR1=
-NETMASK1=
-GATEWAY1=
-NAMESERVER11=
-NAMESERVER12=
