diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/autoexec skell-1.18.modified/etc/rc.d/init.d/autoexec
--- skell-1.18/etc/rc.d/init.d/autoexec	2015-02-06 12:30:47.566458828 +0100
+++ skell-1.18.modified/etc/rc.d/init.d/autoexec	2015-02-06 10:56:56.859921385 +0100
@@ -90,10 +90,6 @@
 	if [ -x $LOCAL_ROOT_DIR/safe_hmi ]; then 
 		sleep 2
 		$LOCAL_ROOT_DIR/safe_hmi > /dev/null 2>&1 &
-
-	elif [ -x $LOCAL_ROOT_DIR/hmi ]; then 
-		sleep 2
-		$LOCAL_ROOT_DIR/hmi -qws > /dev/null 2>&1 &
 	fi
 fi
 
@@ -107,9 +103,9 @@
 
 stop() {
 echo -n "Shutting down 'autoexec'... "
-killall -q keyGenerator factory_test safe_hmi fcrts hmi 
+killall -q keyGenerator factory_test safe_hmi fcrts hmi setup_time ts_calibrate splash
 sleep 1
-killall -9 -q keyGenerator factory_test safe_hmi fcrts hmi 
+killall -9 -q keyGenerator factory_test safe_hmi fcrts hmi setup_time ts_calibrate splash
 dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
 echo "Done."
 }
