diff --exclude CVS --exclude .git -uNr skell-1.18/etc/fstab skell-1.18.modified/etc/fstab
--- skell-1.18/etc/fstab	2011-10-05 09:07:51.352733001 +0200
+++ skell-1.18.modified/etc/fstab	2011-10-05 08:42:33.168735077 +0200
@@ -10,6 +10,7 @@
 devpts          /dev/pts    devpts  gid=5,mode=620    0       0
 sysfs           /sys        sysfs   defaults          0       0
 usbfs        /proc/bus/usb  usbfs   defaults          0       0
+debugfs   /sys/kernel/debug debugfs defaults          0       0
 
 /dev/fd0        /mnt/floppy auto    noauto,owner      0       0
 /dev/cdrom      /mnt/cdrom  iso9660 noauto,owner,ro   0       0
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/inetd.conf skell-1.18.modified/etc/inetd.conf
--- skell-1.18/etc/inetd.conf	2011-10-04 14:38:15.419242000 +0200
+++ skell-1.18.modified/etc/inetd.conf	2011-10-04 14:30:53.670478001 +0200
@@ -23,8 +23,10 @@
 #
 # These are standard services.
 #
-ftp	stream	tcp	nowait	root	/usr/sbin/ftpd	ftpd
-telnet	stream  tcp     nowait  root    /usr/sbin/telnetd	telnetd -i
+ftp	   stream  tcp	nowait	root  /usr/sbin/ftpd	ftpd -w
+telnet stream  tcp  nowait  root  /usr/sbin/telnetd	telnetd -i
+http   stream  tcp  nowait  root  /usr/sbin/httpd   httpd
+
 # nntp	stream	tcp	nowait	root	tcpd	in.nntpd
 # smtp  stream  tcp     nowait  root    tcpd    sendmail -v
 #
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/profile skell-1.18.modified/etc/profile
--- skell-1.18/etc/profile	2011-10-04 14:38:15.443253997 +0200
+++ skell-1.18.modified/etc/profile	2011-10-04 14:35:26.382766055 +0200
@@ -13,3 +13,16 @@
 export PS4='[$LINENO]+'
 
 export GST_PLUGIN_PATH=/usr/lib/fsl_mm_linux/lib/gstreamer-0.10
+
+# ts variable
+export QWS_MOUSE_PROTO=tslib:/dev/input/ts0
+export TSLIB_CONFFILE=/usr/etc/ts.conf
+export TSLIB_PLUGINDIR=/usr/lib/ts
+export TSLIB_TSDEVICE=/dev/input/ts0
+export TSLIB_CONSOLEDEVICE=none
+export TSLIB_CALIBFILE=/local/etc/sysconfig/pointercal
+
+# qt variable
+export QT_QWS_FONTDIR=/usr/lib/fonts
+export POINTERCAL_FILE=$TSLIB_CALIBFILE
+
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/rc.conf skell-1.18.modified/etc/rc.d/rc.conf
--- skell-1.18/etc/rc.d/rc.conf	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/etc/rc.d/rc.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,21 +0,0 @@
-export HOSTNAME=freescale
-export INTERFACE=eth0
-export IPADDR=
-export NETMASK=255.255.255.0
-export BROADCAST=192.168.0.255
-export GATEWAY=192.168.0.1
-export NAMESERVER=192.168.0.1
-export NTP_SERVER=ntp.cs.strath.ac.uk
-export MODLIST=""
-export RAMDIRS=""
-export TMPFS="ramfs"
-export TMPFS_SIZE="512k"
-export READONLY_FS="n"
-export SYSLOG_SOCKET_FILE=""
-export SYSCFG_DHCPC_CMD="udhcpc -b -i"
-
-all_services="hostname filesystems syslog depmod modules network settime inetd portmap dropbear boa dhcpd"
-all_services_r="dhcpd boa dropbear portmap inetd settime network modules depmod syslog filesystems hostname"
-cfg_services="hostname filesystems syslog depmod modules network settime inetd portmap dropbear boa dhcpd"
-cfg_services_r="dhcpd boa dropbear portmap inetd settime network modules depmod syslog filesystems hostname"
-

