diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/autoexec skell-1.18.modified/etc/rc.d/init.d/autoexec
--- skell-1.18/etc/rc.d/init.d/autoexec	2012-02-14 14:13:26.986687764 +0100
+++ skell-1.18.modified/etc/rc.d/init.d/autoexec	2012-02-08 12:47:06.600298005 +0100
@@ -17,7 +17,7 @@
 if [ ! -e $TSLIB_CALIBFILE ]; then
 	echo -n "Starting 'calibration'... "
 	/usr/bin/ts_calibrate
-	sleep 1
+	sync
 	echo "Done."
 fi
 
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/network skell-1.18.modified/etc/rc.d/init.d/network
--- skell-1.18/etc/rc.d/init.d/network	2012-02-14 14:13:26.954687764 +0100
+++ skell-1.18.modified/etc/rc.d/init.d/network	2012-02-14 12:31:57.718687392 +0100
@@ -80,6 +80,10 @@
     if [ "$SYSCFG_IFACE0" = "y" ]
     then
         echo "Setting up networking on $INTERFACE0: "
+        if [ -n "$MAC0" ]
+        then
+			ifconfig $INTERFACE0 hw ether $MAC0
+		fi
         if [ "$IPADDR0" = "dhcp" ]
         then
             if [ "$DEPLOYMENT_STYLE" = "NFS" ]
@@ -107,6 +111,10 @@
     if [ "$SYSCFG_IFACE1" = "y" ]
     then
         echo "Setting up networking on $INTERFACE1: "
+        if [ -n "$MAC1" ]
+        then
+			ifconfig $INTERFACE1 hw ether $MAC1
+		fi
         if [ "$IPADDR1" = "dhcp" ]
         then
             $SYSCFG_DHCPC_CMD $INTERFACE1
diff --exclude CVS --exclude .git -uNr skell-1.18/home/user/.rhosts skell-1.18.modified/home/user/.rhosts
--- skell-1.18/home/user/.rhosts	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/home/user/.rhosts	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-+ +
