diff --exclude CVS --exclude .git -uNr skell-1.18/etc/shadow skell-1.18.modified/etc/shadow
--- skell-1.18/etc/shadow	2012-10-01 16:37:00.297683021 +0200
+++ skell-1.18.modified/etc/shadow	2012-10-01 14:23:56.559550431 +0200
@@ -1,4 +1,4 @@
-root::11851:0:99999:7:::
+root:$1$kmiZb8Op$66xleA/3tYt8y.ktD7H9U1:0:0:99999:7:::
 bin:*:11851:0:99999:7:::
 daemon:*:11851:0:99999:7:::
 adm:*:11851:0:99999:7:::
