diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/filesystems skell-1.18.modified/etc/rc.d/init.d/filesystems
--- skell-1.18/etc/rc.d/init.d/filesystems	2015-04-10 17:39:11.218020878 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/filesystems	2015-04-10 17:32:48.962990422 +0200
@@ -11,10 +11,17 @@
 if [ "$1" = "start" ]
 then
     echo Mounting filesystems
-    if [ "$TMPFS" = "tmpfs" ]
+    if [ `echo $RAMDIRS | grep -c dev` -eq 1 ]
     then
-        mount -n -t $TMPFS shm /dev/shm
+    	echo "dev is a temporary file system"
+	    mkdir -p /dev/shm && mount -n -t $TMPFS shm /dev/shm
+	    mkdir -p /dev/pts && mount -n -t devpts devpts /dev/pts
+    else
+    	echo "dev is not a temporary file system"
+	    mount -n -o remount -w %root% / && mkdir -p /dev/shm
+	    mkdir -p /dev/pts && mount -n -t devpts devpts /dev/pts
     fi
+    
     if [ -n "$TMPFS" ]
     then
         mount -n -t $TMPFS rwfs /mnt/rwfs -o size=$TMPFS_SIZE
@@ -50,10 +57,5 @@
 #    then
 #        rm -f /etc/mtab
 #    fi
-#    ln -s /proc/mounts /etc/mtab
-    if [ ! -d /dev/pts ]
-    then
-        mkdir /dev/pts
-    fi
     mount -a
 fi
