diff -Naur skell-1.18/Makefile skell-1.18-mtl/Makefile
--- skell-1.18/Makefile	2015-07-25 00:38:59.623089706 +0200
+++ skell-1.18-mtl/Makefile	2015-07-30 14:35:19.078791719 +0200
@@ -34,6 +34,7 @@
 	cd $(DESTDIR)/local && ln -s flash/data data
 	cd $(DESTDIR)/local && ln -s flash/etc etc
 	cd $(DESTDIR)/local && ln -s flash/root root
+	cd $(DESTDIR)/local && chmod 0755 flash/root
 	#test -L root || test -d root && rm -rf root
 	rm -rf $(DESTDIR)/root
 	cd $(DESTDIR) && ln -s local/root
