diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/filesystems skell-1.18.modified/etc/rc.d/init.d/filesystems
--- skell-1.18/etc/rc.d/init.d/filesystems	2015-04-16 11:11:06.679052136 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/filesystems	2015-04-16 11:10:38.929184623 +0200
@@ -14,12 +14,24 @@
     if [ `echo $RAMDIRS | grep -c dev` -eq 1 ]
     then
     	echo "dev is a temporary file system"
-	    mkdir -p /dev/shm && mount -n -t $TMPFS shm /dev/shm
-	    mkdir -p /dev/pts && mount -n -t devpts devpts /dev/pts
+	    mkdir -p /dev/shm
+	    mkdir -p /dev/pts
     else
     	echo "dev is not a temporary file system"
-	    mount -n -o remount -w %root% / && mkdir -p /dev/shm
-	    mkdir -p /dev/pts && mount -n -t devpts devpts /dev/pts
+	if [ ! -d /dev/shm -o ! -d /dev/pts ]
+	then
+	    mount -n -o remount -w %root% /
+            mkdir -p /dev/shm
+	    mkdir -p /dev/pts
+	    mount -n -o remount -r %root% /
+	fi
+    fi
+
+    mount -n -t devpts devpts /dev/pts
+
+    if [ "$TMPFS" = "tmpfs" ]
+    then
+        mount -n -t $TMPFS shm /dev/shm
     fi
     
     if [ -n "$TMPFS" ]
