diff --exclude CVS --exclude .git -uNr skell-1.18/etc/mdev.conf skell-1.18.modified/etc/mdev.conf
--- skell-1.18/etc/mdev.conf	2016-03-22 16:34:19.772925578 +0100
+++ skell-1.18.modified/etc/mdev.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,13 +0,0 @@
-# Note: the first field is a regex matcher
-# Syntax: %s %d:%d %s
-# devices user:group mode
-
--$MODALIAS=.*  0:0   660  @modprobe  -s      "$MODALIAS"
-$MODALIAS=.*   0:0   660  $modprobe  -s  -r  "$MODALIAS"
-
-sda[0-9]       0:6   660
-sdb[0-9]       0:6   660
-
-ttyS[0-9]      0:14  640
-tty[0-9]       0:0   660
-tty.+          0:5   660
