diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/autoexec skell-1.18.modified/etc/rc.d/init.d/autoexec
--- skell-1.18/etc/rc.d/init.d/autoexec	2015-04-02 16:32:18.383978985 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/autoexec	2015-03-30 14:04:26.799151965 +0200
@@ -84,11 +84,11 @@
 elif [ -x $LOCAL_ROOT_DIR/fcrts ]
 then
 	echo "Starting applications."
-	dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
+	#dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
 
 	$LOCAL_ROOT_DIR/fcrts > /dev/null 2>&1 & 
 	if [ -x $LOCAL_ROOT_DIR/safe_hmi ]; then 
-		sleep 2
+		#sleep 2
 		$LOCAL_ROOT_DIR/safe_hmi > /dev/null 2>&1 &
 	fi
 fi
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/network skell-1.18.modified/etc/rc.d/init.d/network
--- skell-1.18/etc/rc.d/init.d/network	2015-04-02 16:32:18.379976976 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/network	2015-03-30 14:01:53.459153311 +0200
@@ -81,7 +81,7 @@
         echo "config screen, or directly in $CONFIGFILE."
         echo "IP address setup bypassed"
         echo ""
-        sleep 2
+        #sleep 2
     fi
 
     if [ "$SYSCFG_IFACE0" = "y" ]
diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/settime skell-1.18.modified/etc/rc.d/init.d/settime
--- skell-1.18/etc/rc.d/init.d/settime	2008-09-23 14:48:39.000000000 +0200
+++ skell-1.18.modified/etc/rc.d/init.d/settime	2015-03-30 14:01:30.943151650 +0200
@@ -27,7 +27,7 @@
             echo "    date <mmddhhmnyyyy>"
             echo "    /sbin/hwclock -w"
             touch /etc/setttime_msg
-            sleep 2
+            #sleep 2
         fi
     fi
 fi
