diff -aurN skell-1.18-original/etc/rc.d/init.d/autoexec skell-1.18/etc/rc.d/init.d/autoexec
--- skell-1.18-original/etc/rc.d/init.d/autoexec	2015-10-26 17:07:43.570053382 +0100
+++ skell-1.18/etc/rc.d/init.d/autoexec	2015-10-26 17:15:14.906076165 +0100
@@ -88,7 +88,7 @@
 	#dd if=/dev/zero of=/dev/fb0 >/dev/null 2>&1
 
 	$LOCAL_ROOT_DIR/fcrts > /dev/null 2>&1 & 
-	if [ -x $LOCAL_ROOT_DIR/safe_hmi ]; then 
+	if [ -x $LOCAL_ROOT_DIR/safe_hmi -a ! -e /var/tmp/dont_run_safe_hmi ]; then 
 		#sleep 2
 		$LOCAL_ROOT_DIR/safe_hmi > /dev/null 2>&1 &
 	fi
