diff --exclude CVS --exclude .git -uNr skell-1.18/etc/rc.d/init.d/S10setup skell-1.18.modified/etc/rc.d/init.d/S10setup
--- skell-1.18/etc/rc.d/init.d/S10setup	2012-11-26 12:58:54.210700182 +0100
+++ skell-1.18.modified/etc/rc.d/init.d/S10setup	2012-11-26 12:57:40.106702971 +0100
@@ -19,6 +19,9 @@
     sleep 1
     echo "" | tee $logfile
 	usb_string=`awk '/sd[a-z]: sd[a-z]/ ' < $mntdir/log`
+	if [ -z "$usb_string" ]; then
+		usb_string=`awk '/sd[a-z]:/ ' < $mntdir/log`
+	fi
 
 	echo "Found scsi device: $usb_string" | tee -a $logfile
 
@@ -97,6 +100,7 @@
 
 		if [ -f $mntdir/database.txt ]; then
 			echo " Found the database list file '$mntdir/database.txt'" | tee -a $logfile
+			/usr/bin/dos2unix $mntdir/database.txt 2>&1 | tee -a $logfile
 			if [ -f $mntdir/`cat $mntdir/database.txt` ]; then
 				echo " Found database file '$mntdir/`cat $mntdir/database.txt`'" | tee -a $logfile
 				/bin/cp $mntdir/`cat $mntdir/database.txt` /local/etc/sysconfig/val01.sqlite 2>&1 | tee -a $logfile
@@ -104,7 +108,6 @@
 				echo " Database '$mntdir/`cat $mntdir/database.txt`' updated!" | tee -a $logfile
 			else
 				echo "Error: cannot found database file $mntdir/`cat database.txt`.  Aborting update process."
-				return
 			fi
 		else
         	if [ -f $mntdir/val01.sqlite ]; then
