diff --exclude CVS --exclude .git -uNr skell-1.18/Makefile skell-1.18.modified/Makefile
--- skell-1.18/Makefile	2014-12-19 15:24:32.770186107 +0100
+++ skell-1.18.modified/Makefile	2014-12-16 16:24:53.760288125 +0100
@@ -32,6 +32,13 @@
 	rm -rf home
 	mkdir -p local/home
 	ln -s local/home
+	mkdir -p local/etc/sysconfig/ppp
+	rm -rf etc/ppp
+	ln -s /local/etc/sysconfig/ppp etc/ppp
+	mkdir -p local/etc/sysconfig/chatscripts
+	rm -rf etc/chatscripts
+	ln -s /local/etc/sysconfig/chatscripts etc/chatscripts
+
 	cd $(RPM_BUILD_DIR)/skell-1.18
 	#cd -
 
