--- ltib/dist/lfs-5.1/sysconfig/sysconfig-mx.spec	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/sysconfig/sysconfig-mx.spec	2015-10-25 10:21:38.284137205 +0100
@@ -8,14 +8,25 @@
 Vendor		: Freescale
 Packager	: Stuart Hughes
 Group		: System Environment/Base
+Source          : inittab
 BuildRoot	: %{_tmppath}/%{name}
 Prefix		: %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed sysconfig for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+sysconfig package contents restricted to just the files that
+are needed at run time on the target.
+
 
 %Description
 %{summary}
 
 %Prep
-#%setup
 
 %Build
 
@@ -63,6 +74,36 @@
 then
     boa=boa
 fi
+if [ "$SYSCFG_START_VSFTPD" = "y" ]
+then
+    vsftpd=vsftpd
+fi
+if [ "$SYSCFG_START_TELNET" = "y" ]
+then
+    telnetd=telnetd
+	mkdir -p $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/init.d
+cat <<'EOF' > $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/init.d/telnetd
+#!/bin/sh
+
+if [ ! -x /usr/sbin/telnetd ]
+then
+	exit 0
+fi
+
+if [ "$1" = "stop" -o "$1" = "restart" ]
+then
+	echo "Stopping the telnet daemon: "
+	killall telnetd
+fi
+
+if [ "$1" = "start" -o "$1" = "restart" ]
+then
+	echo "Starting the telnet daemon: "
+	/usr/sbin/telnetd
+fi
+EOF
+	chmod +x $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/init.d/telnetd
+fi
 if [ "$SYSCFG_SETTIME" = "y" ]
 then
     settime=settime
@@ -91,14 +132,24 @@
 then
         fslgnome=fslgnome
 fi
+if [ "$SYSCFG_START_AUTOEXEC" = "y" ]
+then
+        autoexec=autoexec
+	sdcheck=sdcheck
+	usbmode=usbmode
+fi
 
+if [ "$SYSCFG_START_S10SETUP" = "y" ]
+then
+        S10setup=S10setup
+fi
 cat <<EOF > $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc.conf
-all_services="mount-proc-sys mdev udev hostname devfsd depmod modules filesystems syslog network inetd portmap dropbear sshd boa smb dhcpd settime fslgnome watchdog bluetooth gtk2 pango"
-all_services_r="pango gtk2 bluetooth watchdog fslgnome settime dhcpd smb boa sshd dropbear portmap inetd network syslog filesystems modules depmod devfsd hostname udev mdev mount-proc-sys"
+all_services="mount-proc-sys mdev udev hostname devfsd depmod modules filesystems sdcheck syslog network inetd portmap dropbear sshd boa vsftpd smb dhcpd settime fslgnome watchdog bluetooth gtk2 pango telnetd usbmode S10setup autoexec"
+all_services_r="autoexec S10setup usbmode telnetd pango gtk2 bluetooth watchdog fslgnome settime dhcpd smb boa vsftpd sshd dropbear portmap inetd network syslog sdcheck filesystems modules depmod devfsd hostname udev mdev mount-proc-sys"
 
-cfg_services="mount-proc-sys $mdev $udev hostname $devfsd depmod modules filesystems $syslog $network $inetd $portmap $dropbear $sshd $boa $smb $dhcpd $settime $fslgnome $watchdog $bluetooth $gtk2 $pango"
+cfg_services="mount-proc-sys $mdev $udev hostname $devfsd depmod modules filesystems $sdcheck $usbmode $autoexec $syslog $network $inetd $portmap $dropbear $sshd $boa $vsftpd $smb $dhcpd $settime $fslgnome $watchdog $bluetooth $gtk2 $pango $telnetd $S10setup"
 
-cfg_services_r="$pango $gtk2 $bluetooth $watchdog $fslgnome $settime $dhcpd $smb $boa $sshd $dropbear $portmap $inetd $network $syslog filesystems modules depmod $devfsd hostname $udev $mdev mount-proc-sys"
+cfg_services_r="$autoexec $S10setup $usbmode $pango $gtk2 $bluetooth $watchdog $fslgnome $settime $dhcpd $smb $boa $vsftpd $sshd $dropbear $portmap $inetd $network $syslog $sdcheck filesystems modules depmod $devfsd hostname $udev $mdev mount-proc-sys $telnetd"
 
 export HOSTNAME="${SYSCFG_HOSTNAME:-$PLATFORM}"
 export NTP_SERVER="$SYSCFG_NTP_SERVER"
@@ -154,61 +205,7 @@
     else
 	sys_login="::respawn:-/bin/sh"
     fi
-
-    cat <<EOF > $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc_mxc.S
-#!/bin/bash
-#
-if grep -sq ttymxc0 /proc/cmdline; then
-	/sbin/getty -L ttymxc0 115200 vt100
-elif grep -sq ttymxc1 /proc/cmdline; then
-	/sbin/getty -L ttymxc1 115200 vt100
-elif grep -sq ttymxc2 /proc/cmdline; then
-	/sbin/getty -L ttymxc2 115200 vt100
-elif grep -sq ttymxc3 /proc/cmdline; then
-	/sbin/getty -L ttymxc3 115200 vt100
-else
-	sleep 100000
-fi
-EOF
-chmod +x $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc_mxc.S
-
-cat <<EOF > $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc_gpu.S
-#!/bin/bash
-CPUREV=\$(cat /proc/cpuinfo | grep Revision | awk '{print \$3}' | awk '{print substr(\$0,1,2)}')
-FILEVG=/usr/lib/libOpenVG.so
-FILEVG3D=/usr/lib/libOpenVG_3D.so
-FILEVG355=/usr/lib/libOpenVG_355.so
-if  [ -e \$FILEVG3D ] && [ -e \$FILEVG355 ]
-then
-  if  [ \$CPUREV == "61" ] || [ \$CPUREV == "63" ] || [ \$CPUREV == "60" ] && [ -e  \$FILEVG ]
-  then
-        rm -f \$FILEVG
-  fi
-  if [ \$CPUREV == "61" ]
-  then
-        ln -s \$FILEVG3D \$FILEVG
-  fi
-  if [ \$CPUREV == "63" ]
-  then
-        ln -s \$FILEVG355 \$FILEVG
-  fi
-  if [ \$CPUREV == "60" ]
-  then
-        ln -s \$FILEVG355 \$FILEVG
-  fi
-fi
-EOF
-chmod +x $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc_gpu.S
-
-    cat <<EOF > $RPM_BUILD_ROOT/%{pfx}/etc/inittab
-# see busybox-1.00rc2/examples/inittab for more examples
-::sysinit:/etc/rc.d/rcS
-$sys_login
-::sysinit:/etc/rc.d/rc_gpu.S
-::ctrlaltdel:/sbin/reboot
-::shutdown:/etc/rc.d/rcS stop
-::restart:/sbin/init
-EOF
+    install -m 644 %{SOURCE0} $RPM_BUILD_ROOT/%{pfx}/etc/inittab
 else
     # SysVInit
     if [ "$SYSCFG_WANT_LOGIN_TTY" = "y" ]
@@ -242,18 +239,20 @@
 #
 # This script will be executed *after* all the other init scripts.
 # You can put your own initialization stuff in here
+
+# Rebuild the RPM db on target at first boot.
 if [ -x "/usr/bin/rpm" -a -e "/tmp/ltib" ]
 then
-    echo "rebuilding rpm database"
+    echo "rebuilding the RPM database"
     rm -rf /tmp/ltib
     rpm --rebuilddb
 fi
 
-# fix up permissions
-if [ -d /home/user ]
-then
-    chown -R user.user /home/user
-fi
+# Fix the user home directory permissions.
+test -d /home/user && chown -R user.user /home/user
+
+# Fix the root home directory permissions.
+test -d /local/flash/root && chmod 0755 /local/flash/root
 
 # Add nodes when running under the hypervisor and static devices
 if [ -r /sys/class/misc/fsl-hv/dev -a ! -r /dev/fsl-hv ]
@@ -287,30 +286,34 @@
 fi
 
 for i in 0 1 2; do
-    if [ -e /sys/class/graphics/fb$i ]; then
-        chmod 0666 /sys/class/graphics/fb$i/pan
-    fi
+	test -e /sys/class/graphics/fb$i && chmod 0666 /sys/class/graphics/fb$i/pan
 done
-
 EOF
-chmod +x $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc.local
+chmod a+x $RPM_BUILD_ROOT/%{pfx}/etc/rc.d/rc.local
 
 cat <<EOF > $RPM_BUILD_ROOT/%{pfx}/etc/issue
-$( gcc --version | grep 'gcc')
+
+$(gcc --version | grep gcc)
 root filesystem built on $(date -R)
-Freescale Semiconductor, Inc.
+MECT s.r.l. http://www.mect.it/
 
 EOF
 
 # The kernel attempts to run /init (not /sbin/init!) from initramfs images
-if [ "$SYSCFG_DEPLOYMENT_STYLE" = "INITRAMFS" ]
-then
-    ln -s /sbin/init $RPM_BUILD_ROOT/%{pfx}/init
-fi
+test "$SYSCFG_DEPLOYMENT_STYLE" = "INITRAMFS" && ln -s /sbin/init $RPM_BUILD_ROOT/%{pfx}/init
+
+cd %{_builddir}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed -n '
+	p;
+' all_files.lst > rfs_files.lst
 
 %Clean
 rm -rf $RPM_BUILD_ROOT
 
-%Files
+%Files -f all_files.lst
+%defattr(-,root,root)
+
+
+%Files rfs -f rfs_files.lst
 %defattr(-,root,root)
-%{pfx}/*
