diff --exclude CVS --exclude .git -uNr tslib-1.0/plugins/input-raw.c tslib-1.0.modified/plugins/input-raw.c
--- tslib-1.0/plugins/input-raw.c	2014-11-28 19:09:21.016990440 +0100
+++ tslib-1.0.modified/plugins/input-raw.c	2014-11-28 18:24:23.388990809 +0100
@@ -53,7 +53,7 @@
 
 	if (! ((ioctl(ts->fd, EVIOCGVERSION, &version) >= 0) &&
 		//original (version == EV_VERSION) &&
-		(version == (EV_VERSION -1)) &&
+		((version == (EV_VERSION -1)) || (version == EV_VERSION)) &&
 		(ioctl(ts->fd, EVIOCGBIT(0, sizeof(bit) * 8), &bit) >= 0) &&
 		(bit & (1 << EV_ABS)) &&
 		(ioctl(ts->fd, EVIOCGBIT(EV_ABS, sizeof(absbit) * 8), &absbit) >= 0) &&

