--- ltib/dist/lfs-5.1/u-boot/u-boot-common.tmpl	2013-01-09 04:58:38.000000000 +0100
+++ ltib.new/dist/lfs-5.1/u-boot/u-boot-common.tmpl	2015-10-12 10:46:10.458469000 +0200
@@ -18,7 +18,8 @@
     make HOSTCC="$BUILDCC" CROSS_COMPILE=$TOOLCHAIN_PREFIX O=$BUILD_DIR distclean
 fi
 make HOSTCC="$BUILDCC" CROSS_COMPILE=$TOOLCHAIN_PREFIX O=$BUILD_DIR $PKG_U_BOOT_CONFIG_TYPE
-make HOSTCC="$BUILDCC" HOSTSTRIP="$BUILDSTRIP" \
+make -j`grep -c ^processor /proc/cpuinfo` \
+     HOSTCC="$BUILDCC" HOSTSTRIP="$BUILDSTRIP" \
      CROSS_COMPILE=$TOOLCHAIN_PREFIX $PKG_U_BOOT_BUILD_ARGS \
      O=$BUILD_DIR all
 
