--- ltib/config/platform/imx/u-boot.spec.in	2013-01-09 04:58:38.000000000 +0100
+++ ltib.modified/config/platform/imx/u-boot.spec.in	2015-08-15 23:40:18.935646832 +0200
@@ -13,6 +13,7 @@
 Source          : %{name}-%{version}.tar.bz2
 Source1         : %{name}-v%{version}-%{release}.tar.bz2
 BuildRoot       : %{_tmppath}/%{name}
+Patch1          : u-boot-2009.08-1418123257.patch
 Prefix          : %{pfx}
 
 %Description
@@ -28,5 +29,6 @@
 
 chmod 755 patches/patch-uboot.sh
 ./patches/patch-uboot.sh
+%patch1 -p1
 
 
