--- ltib/dist/lfs-5.1/xenomai/xenomai.spec	1970-01-01 01:00:00.000000000 +0100
+++ ltib.new/dist/lfs-5.1/xenomai/xenomai.spec	2015-10-12 10:46:10.686466000 +0200
@@ -0,0 +1,60 @@
+%define pfx /opt/freescale/rootfs/%{_target_cpu}
+
+Summary         : xenomai - Real Time extension for the Linux kernel
+Name            : xenomai
+Version         : 2.6.0
+Release         : 1
+License         : GPL
+Vendor          : Freescale
+Packager        : Mect
+Group           : System Environment
+Source          : %{name}-%{version}.tar.bz2
+BuildRoot       : %{_tmppath}/%{name}
+Prefix          : %{pfx}
+AutoReqProv     : no
+
+
+%Package rfs
+Summary         : Trimmed xenomai for root file system requirements.
+Group           : Applications/File
+AutoReqProv     : no
+%Description rfs
+xenomai package contents restricted to just the files that
+are needed at run time on the target.
+
+
+%Description
+%{summary}
+
+#The Prep target is performed together with the kernel if kernel with ipipe is selected.
+#  %{name}-%{version}
+
+%Prep
+%setup
+./configure CFLAGS="-march=armv5te" LDFLAGS="-march=armv5te" --build=%{_build} --host=$CFGHOST
+
+%Build
+make DESTDIR=$RPM_BUILD_ROOT%{pfx} all
+
+%Install
+rm -rf $RPM_BUILD_ROOT
+make DESTDIR=$RPM_BUILD_ROOT%{pfx} install
+
+cd %{_builddir}/%{name}-%{version}
+( cd $RPM_BUILD_ROOT; find .%{pfx} \( \( -not -type d \) -o \( -type d -empty \) \) -print | sed 's/^\.//' ) > all_files.lst
+sed '
+    \,^%{pfx}%{_prefix}/xenomai/share/doc, d;
+    \,^%{pfx}%{_prefix}/xenomai/share/man, d;
+    \,^%{pfx}%{_prefix}/xenomai/include, d;
+    /\.l\?a$/ d;
+' all_files.lst > rfs_files.lst
+
+%Clean
+rm -rf $RPM_BUILD_ROOT
+
+%Files -f all_files.lst
+%defattr(-,root,root)
+
+
+%Files rfs -f rfs_files.lst
+%defattr(-,root,root)
